<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interoperability: SpatialExperiment • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Interoperability: SpatialExperiment">
<meta property="og:description" content="Giotto">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/errorsfaqsandtips.html">Solutions to Installation Errors</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html">Spatial geometries</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Interoperability: SpatialExperiment</h1>
                        <h4 data-toc-skip class="author">Iqra Amin,
Irzam Sarfraz</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/HEAD/vignettes/interoperability_spatialexperiment.Rmd" class="external-link"><code>vignettes/interoperability_spatialexperiment.Rmd</code></a></small>
      <div class="hidden name"><code>interoperability_spatialexperiment.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Giotto allows seamless analysis of spatial transcriptomics data
provided through a range of functions and algorithms. At the same time,
it also allows effortless conversion to other object types, so the data
can be analyzed with other tools including packages from Bioconductor
that mostly use *Experiment objects and their derivatives. In this
tutorial, we show how cross-conversion between Giotto and
SpatialExperiment objects can complement spatial transcriptomic
analyses.</p>
<p>In the first part of the tutorial, we load an example Giotto object,
convert it to SpatialExperiment object and use a multitude of packages
from Bioconductor to analyze it. In the second part, we use an example
SpatialExperiment object, convert it to Giotto and drive the analysis
with the toolkit.</p>
</div>
<div class="section level2">
<h2 id="giottoobj-to-spatialexperiment">1.0 Giottoobj to SpatialExperiment<a class="anchor" aria-label="anchor" href="#giottoobj-to-spatialexperiment"></a>
</h2>
<p>In this part of the tutorial, we show the conversion of a an example
Giotto object to SpatialExperiment object and use several packages to
process it. Initially, we load the required packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/">Giotto</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/GiottoData/">GiottoData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-a-giotto-object">1.2 Load a Giotto Object<a class="anchor" aria-label="anchor" href="#load-a-giotto-object"></a>
</h2>
<p>First, we load a example dataset from the GiottoData package. For
this tutorial, we import a mini mouse brain dataset generated using the
visium protocol.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/loadGiottoMini.html">loadGiottoMini</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">"visium"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: pandas was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: igraph was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: leidenalg was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: community was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: networkx was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: sklearn was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: pandas was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: igraph was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: leidenalg was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: community was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: networkx was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: sklearn was not found with python path: /usr/bin/python3</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="convert-to-a-spatialexperiment-obj">1.3 Convert to a SpatialExperiment obj<a class="anchor" aria-label="anchor" href="#convert-to-a-spatialexperiment-obj"></a>
</h2>
<p>We now convert this imported giotto object into a SpatialExperiment
object using the “giottoToSpatialExperiment” function.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gspe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giottoToSpatialExperiment.html">giottoToSpatialExperiment</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span></span></code></pre></div>
<p>For each spatial unit the converter function returns a separate
SpatialExperiment object. The output is a list of objects (one object
for each unit) therefore we select the first object and use it
downstream.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">gspe</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="downstream-analysis-using-spatialexperiment-object">1.4 Downstream Analysis using SpatialExperiment object<a class="anchor" aria-label="anchor" href="#downstream-analysis-using-spatialexperiment-object"></a>
</h2>
<p>Now, we do some processing on the object including quality control,
normalization, feature selection and dimensionality reduction.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/ggspavis" class="external-link">ggspavis</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="quality-control-metrics">1.4.1 Quality Control Metrics<a class="anchor" aria-label="anchor" href="#quality-control-metrics"></a>
</h3>
<p>Using the <em>scater</em> package from Bioconductor, we perform some
initial quality control analysis. We particularly look for mitochondrial
genes and compute some quality control metrics including total counts
and features per cell. We then visualize these metrics and select
thresholds to remove low quality cells.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># identify mitochondrial genes</span></span>
<span><span class="va">is_mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"(^MT-)|(^mt-)"</span>, <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">is_mito</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## is_mito</span></span>
<span><span class="co">## FALSE </span></span>
<span><span class="co">##   634</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate per-spot QC metrics and store in colData</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html" class="external-link">addPerCellQC</a></span><span class="op">(</span><span class="va">spe</span>, subsets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>mito <span class="op">=</span> <span class="va">is_mito</span><span class="op">)</span>,assay.type <span class="op">=</span> <span class="st">"raw_rna_cell"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 14 columns</span></span>
<span><span class="co">##                               cell_ID in_tissue  nr_feats perc_feats total_expr</span></span>
<span><span class="co">##                           &lt;character&gt; &lt;integer&gt; &lt;integer&gt;  &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">## AAAGGGATGTAGCAAG-1 AAAGGGATGTAGCAAG-1         1       227    35.8044    995.149</span></span>
<span><span class="co">## AAATGGCATGTCTTGT-1 AAATGGCATGTCTTGT-1         1       283    44.6372   1123.875</span></span>
<span><span class="co">## AAATGGTCAATGTGCC-1 AAATGGTCAATGTGCC-1         1       248    39.1167   1027.632</span></span>
<span><span class="co">## AAATTAACGGGTAGCT-1 AAATTAACGGGTAGCT-1         1       222    35.0158    974.377</span></span>
<span><span class="co">## AACAACTGGTAGTTGC-1 AACAACTGGTAGTTGC-1         1       313    49.3691   1174.488</span></span>
<span><span class="co">## AACAGGAAATCGAATA-1 AACAGGAAATCGAATA-1         1       338    53.3123   1229.935</span></span>
<span><span class="co">##                    leiden_clus custom_leiden   sample_id       sum  detected</span></span>
<span><span class="co">##                      &lt;numeric&gt;     &lt;numeric&gt; &lt;character&gt; &lt;numeric&gt; &lt;integer&gt;</span></span>
<span><span class="co">## AAAGGGATGTAGCAAG-1           3             1    sample01       470       227</span></span>
<span><span class="co">## AAATGGCATGTCTTGT-1           2             3    sample01       843       283</span></span>
<span><span class="co">## AAATGGTCAATGTGCC-1           5             2    sample01       725       248</span></span>
<span><span class="co">## AAATTAACGGGTAGCT-1           4             2    sample01       529       222</span></span>
<span><span class="co">## AACAACTGGTAGTTGC-1           4             2    sample01      1023       313</span></span>
<span><span class="co">## AACAGGAAATCGAATA-1           2             3    sample01      1403       338</span></span>
<span><span class="co">##                    subsets_mito_sum subsets_mito_detected subsets_mito_percent</span></span>
<span><span class="co">##                           &lt;numeric&gt;             &lt;integer&gt;            &lt;numeric&gt;</span></span>
<span><span class="co">## AAAGGGATGTAGCAAG-1                0                     0                    0</span></span>
<span><span class="co">## AAATGGCATGTCTTGT-1                0                     0                    0</span></span>
<span><span class="co">## AAATGGTCAATGTGCC-1                0                     0                    0</span></span>
<span><span class="co">## AAATTAACGGGTAGCT-1                0                     0                    0</span></span>
<span><span class="co">## AACAACTGGTAGTTGC-1                0                     0                    0</span></span>
<span><span class="co">## AACAGGAAATCGAATA-1                0                     0                    0</span></span>
<span><span class="co">##                        total</span></span>
<span><span class="co">##                    &lt;numeric&gt;</span></span>
<span><span class="co">## AAAGGGATGTAGCAAG-1       470</span></span>
<span><span class="co">## AAATGGCATGTCTTGT-1       843</span></span>
<span><span class="co">## AAATGGTCAATGTGCC-1       725</span></span>
<span><span class="co">## AAATTAACGGGTAGCT-1       529</span></span>
<span><span class="co">## AACAACTGGTAGTTGC-1      1023</span></span>
<span><span class="co">## AACAGGAAATCGAATA-1      1403</span></span></code></pre>
<p>We plot a histogram of library size and select a threshold of 50 to
remove low quality cells.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># histogram of library sizes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">sum</span>, breaks <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/lib-size-1.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select QC threshold</span></span>
<span><span class="va">qc_lib_size</span> <span class="op">&lt;-</span> <span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">sum</span> <span class="op">&lt;</span> <span class="fl">50</span></span></code></pre></div>
<p>We select a threshold of 18 for number of features detected per cell
and remove the low quality cells.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># histogram of numbers of expressed genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span>, breaks <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/filtering-1.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select QC threshold for number of expressed genes</span></span>
<span><span class="va">qc_detected</span> <span class="op">&lt;-</span> <span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span> <span class="op">&lt;</span> <span class="fl">18</span></span>
<span><span class="co"># combined set of discarded spots</span></span>
<span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">qc_lib_size</span> <span class="op">|</span> <span class="va">qc_detected</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">discard</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## discard</span></span>
<span><span class="co">## FALSE </span></span>
<span><span class="co">##   624</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">discard</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="op">!</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">discard</span><span class="op">]</span></span>
<span><span class="va">spe</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SpatialExperiment </span></span>
<span><span class="co">## dim: 634 624 </span></span>
<span><span class="co">## metadata(1): image</span></span>
<span><span class="co">## assays(3): raw_rna_cell normalized_rna_cell scaled_rna_cell</span></span>
<span><span class="co">## rownames(634): Gna12 Ccnd2 ... 9630013A20Rik 2900040C04Rik</span></span>
<span><span class="co">## rowData names(7): feat_ID nr_cells ... mean_expr_det hvf</span></span>
<span><span class="co">## colnames(624): AAAGGGATGTAGCAAG-1 AAATGGCATGTCTTGT-1 ...</span></span>
<span><span class="co">##   TTGTGGCCCTGACAGT-1 TTGTTCAGTGTGCTAC-1</span></span>
<span><span class="co">## colData names(15): cell_ID in_tissue ... total discard</span></span>
<span><span class="co">## reducedDimNames(5): pca custom_pca umap custom_umap tsne</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## spatialCoords names(2) : sdimx sdimy</span></span>
<span><span class="co">## imgData names(1): sample_id</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="normalization">1.4.2 Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h3>
<p>Now we calculate log-transformed normalized counts(logcounts) using
library size factors using the scran package from Bioconductor.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># calculate library size factors</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/librarySizeFactors.html" class="external-link">computeLibraryFactors</a></span><span class="op">(</span><span class="va">spe</span>, assay.type<span class="op">=</span><span class="st">"raw_rna_cell"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">sizeFactors</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##  0.1032  0.5003  0.8066  1.0000  1.2177  4.6123</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">sizeFactors</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/normalization-1.png" width="700"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate logcounts and store in object</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">spe</span>, assay.type<span class="op">=</span><span class="st">"raw_rna_cell"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check</span></span>
<span><span class="fu">assayNames</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "raw_rna_cell"        "normalized_rna_cell" "scaled_rna_cell"    </span></span>
<span><span class="co">## [4] "logcounts"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="feature-selection">1.4.3 Feature selection<a class="anchor" aria-label="anchor" href="#feature-selection"></a>
</h3>
<p>To identify HVGs (highly variable genes), we use a mean-variance
method “modelGeneVar” from the scran package.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># fit mean-variance relationship</span></span>
<span><span class="va">dec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html" class="external-link">modelGeneVar</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="co"># visualize mean-variance relationship</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">metadata</span><span class="op">(</span><span class="va">dec</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit</span><span class="op">$</span><span class="va">mean</span>, <span class="va">fit</span><span class="op">$</span><span class="va">var</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"mean of log-expression"</span>, ylab <span class="op">=</span> <span class="st">"variance of log-expression"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/curve.html" class="external-link">curve</a></span><span class="op">(</span><span class="va">fit</span><span class="op">$</span><span class="fu">trend</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"dodgerblue"</span>, add <span class="op">=</span> <span class="cn">TRUE</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/hvg-detection-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="dimentionality-reduction-pca">1.4.4 Dimentionality Reduction (PCA)<a class="anchor" aria-label="anchor" href="#dimentionality-reduction-pca"></a>
</h3>
<p>Now we apply dimensionality reduction methods to visualize the data
and to generate inputs for further downstream analyses. For this
purpose, we use the runPCA method to compute the top principal
components and then finally plot the top two components.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># compute PCA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDimNames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "pca"         "custom_pca"  "umap"        "custom_umap" "tsne"       </span></span>
<span><span class="co">## [6] "PCA"</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot top 2 PCA dimensions</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotDimRed.html" class="external-link">plotDimRed</a></span><span class="op">(</span><span class="va">spe</span> , type <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/pca-1.png" width="700"></p>
<p>To visualize the clusters, we use a non-linear method “UMAP”, perform
clustering and plot the cluster labels over on the UMAP.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html" class="external-link">runUMAP</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">spe</span>,<span class="st">"UMAP"</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>,<span class="st">"UMAP2"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotDimRed.html" class="external-link">plotDimRed</a></span><span class="op">(</span><span class="va">spe</span>, type <span class="op">=</span> <span class="st">"UMAP"</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/umap-1.png" width="700"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># graph-based clustering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">k</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">gGraphsnn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html" class="external-link">buildSNNGraph</a></span><span class="op">(</span><span class="va">spe</span>, k <span class="op">=</span> <span class="va">k</span>, use.dimred <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span></span>
<span><span class="va">g_walk</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/cluster_walktrap.html" class="external-link">cluster_walktrap</a></span><span class="op">(</span><span class="va">gGraphsnn</span><span class="op">)</span></span>
<span><span class="va">clus</span> <span class="op">&lt;-</span> <span class="va">g_walk</span><span class="op">$</span><span class="va">membership</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">clus</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## clus</span></span>
<span><span class="co">##   1   2   3   4   5   6   7   8   9  10  11  12 </span></span>
<span><span class="co">##  93  93  24 112  49  34  59  21  30  78  14  17</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot clusters in UMAP reduced dimensions</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="va">clus</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotDimRed.html" class="external-link">plotDimRed</a></span><span class="op">(</span><span class="va">spe</span>, type <span class="op">=</span> <span class="st">"UMAP"</span>,annotate <span class="op">=</span> <span class="st">"label"</span>, palette <span class="op">=</span> <span class="st">"libd_layer_colors"</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/umap-2.png" width="700"></p>
<p>We identify the marker genes in each cluster and visualize the top
genes in each cluster using a pheatmap</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># test for marker genes</span></span>
<span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMarkers.html">findMarkers</a></span><span class="op">(</span><span class="va">spe</span>, test <span class="op">=</span> <span class="st">"binom"</span>, direction <span class="op">=</span> <span class="st">"up"</span><span class="op">)</span></span>
<span><span class="co"># plot log-fold changes for one cluster over all other clusters</span></span>
<span><span class="co"># selecting cluster 1</span></span>
<span><span class="va">interesting</span> <span class="op">&lt;-</span> <span class="va">markers</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">best_set</span> <span class="op">&lt;-</span> <span class="va">interesting</span><span class="op">[</span><span class="va">interesting</span><span class="op">$</span><span class="va">Top</span> <span class="op">&lt;=</span> <span class="fl">5</span>, <span class="op">]</span></span>
<span><span class="va">logFCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getMarkerEffects.html" class="external-link">getMarkerEffects</a></span><span class="op">(</span><span class="va">best_set</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">logFCs</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>, <span class="fl">5</span>, length.out <span class="op">=</span> <span class="fl">101</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/marker-detection-1.png" width="700"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot log-transformed normalized expression of top genes for one cluster</span></span>
<span><span class="va">top_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">rownames</a></span><span class="op">(</span><span class="va">interesting</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html" class="external-link">plotExpression</a></span><span class="op">(</span><span class="va">spe</span>, x <span class="op">=</span> <span class="st">"label"</span>, features <span class="op">=</span> <span class="va">top_genes</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/marker-detection-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="spatialexperiment-to-giottoobj">2.0 SpatialExperiment to Giottoobj<a class="anchor" aria-label="anchor" href="#spatialexperiment-to-giottoobj"></a>
</h2>
<p>In this section, we load a sample dataset in the form of a
SpatialExperiment object, convert it to Giotto and drive the downstream
analysis with the Giotto toolkit.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/STexampleData" class="external-link">STexampleData</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-a-sample-spatialexperiment-obj">2.1 Load a sample SpatialExperiment obj<a class="anchor" aria-label="anchor" href="#load-a-sample-spatialexperiment-obj"></a>
</h2>
<p>Loading sample SpatialExperiment Object as spe</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/STexampleData.html" class="external-link">Visium_humanDLPFC</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">spe</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SpatialExperiment </span></span>
<span><span class="co">## dim: 33538 4992 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span></span>
<span><span class="co">##   ENSG00000268674</span></span>
<span><span class="co">## rowData names(3): gene_id gene_name feature_type</span></span>
<span><span class="co">## colnames(4992): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...</span></span>
<span><span class="co">##   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1</span></span>
<span><span class="co">## colData names(7): barcode_id sample_id ... ground_truth cell_count</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span><span class="co">## imgData names(4): sample_id image_id data scaleFactor</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="convert-to-a-giotto-obj">2.2 Convert to a giotto obj<a class="anchor" aria-label="anchor" href="#convert-to-a-giotto-obj"></a>
</h2>
<p>Converting SpatialExperiment Object to giottoobj. The new
giottoobject is n_g.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#create giottoObject from spe</span></span>
<span><span class="va">n_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/spatialExperimentToGiotto.html">spatialExperimentToGiotto</a></span><span class="op">(</span><span class="va">spe</span>, python_path <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/instructions-generic.html">instructions</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">python_path</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##   image  has already been used, will be overwritten</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="downstream-analysis-with-the-giotto-toolkit">2.3 Downstream analysis with the giotto toolkit<a class="anchor" aria-label="anchor" href="#downstream-analysis-with-the-giotto-toolkit"></a>
</h2>
<p>We filter the giottoobj and then run normalization on it.</p>
<div class="section level3">
<h3 id="high-efficiency-data-processing">2.3.1 High Efficiency Data Processing<a class="anchor" aria-label="anchor" href="#high-efficiency-data-processing"></a>
</h3>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/processGiotto.html">processGiotto</a></span><span class="op">(</span></span>
<span>    <span class="va">n_g</span>,</span>
<span>    filter_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                         feat_det_in_min_cells <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                         min_det_feats_per_cell <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    norm_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>norm_methods <span class="op">=</span> <span class="st">'standard'</span>,</span>
<span>                       scale_feats <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                       scalefactor <span class="op">=</span> <span class="fl">6000</span><span class="op">)</span>,</span>
<span>    stat_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_values <span class="op">=</span> <span class="st">'normalized'</span><span class="op">)</span>,</span>
<span>    adjust_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'normalized'</span><span class="op">)</span>,</span>
<span>                         covariate_columns <span class="op">=</span> <span class="st">'nr_feats'</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1. start filter step </span></span>
<span><span class="co">## completed 1: preparation</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Unknown argument 'on' has been passed.</span></span></code></pre>
<pre><code><span><span class="co">## completed 2: subset expression data </span></span>
<span><span class="co">## completed 3: subset spatial locations </span></span>
<span><span class="co">## completed 4: subset cell metadata </span></span>
<span><span class="co">## completed 5: subset feature metadata </span></span>
<span><span class="co">## completed 6: subset spatial network(s) </span></span>
<span><span class="co">## completed 7: subsetted dimension reductions </span></span>
<span><span class="co">## completed 8: subsetted nearest network(s) </span></span>
<span><span class="co">## completed 9: subsetted spatial enrichment results</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: pandas was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: igraph was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: leidenalg was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: community was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: networkx was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## Warning in .check_giotto_python_modules(my_python_path = instructions(.Object, : module: sklearn was not found with python path: /usr/bin/python3</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Feature type:  rna </span></span>
<span><span class="co">## Number of cells removed:  0  out of  4992 </span></span>
<span><span class="co">## Number of feats removed:  21922  out of  33538 </span></span>
<span><span class="co">## 2. start normalization step </span></span>
<span><span class="co">## 3. start cell and gene statistics step </span></span>
<span><span class="co">## 4. start adjusted matrix step</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="dimension-reduction-and-pca">2.3.2 Dimension Reduction and PCA<a class="anchor" aria-label="anchor" href="#dimension-reduction-and-pca"></a>
</h3>
<p>We first find the highly variable features (HVF) and then use these
to compute the reduced dimensions using PCA.</p>
<p>Calculate HVF using the loess regression prediction model:</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">n_g</span>, method <span class="op">=</span> <span class="st">'cov_loess'</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/giotto-hvf-1.png" width="700"></p>
<pre><code><span><span class="co">## return_plot = TRUE and return_gobject = TRUE </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           plot will not be returned to object, but can still be saved with save_plot = TRUE or manually</span></span></code></pre>
<p>PCA can be run based on the highly variable genes. After PCA, a tSNE,
a UMAP, or both may be run. For this example, highly variable genes have
been identified using Loess Regression predictions.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Select genes highly variable genes that fit specified statistics</span></span>
<span><span class="co"># These are both found within feature metadata</span></span>
<span><span class="va">feature_metadata</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/getFeatureMetadata.html">getFeatureMetadata</a></span><span class="op">(</span><span class="va">n_g</span><span class="op">)</span><span class="op">[</span><span class="op">]</span></span>
<span><span class="va">featgenes</span> <span class="op">=</span> <span class="va">feature_metadata</span><span class="op">[</span><span class="va">hvf</span> <span class="op">==</span> <span class="st">'yes'</span> <span class="op">&amp;</span> <span class="va">perc_cells</span> <span class="op">&gt;</span> <span class="fl">4</span> <span class="op">&amp;</span> <span class="va">mean_expr_det</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">]</span><span class="op">$</span><span class="va">feat_ID</span></span>
<span></span>
<span><span class="co">## run PCA on expression values (default)</span></span>
<span><span class="va">n_g</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">n_g</span>, feats_to_use <span class="op">=</span> <span class="va">featgenes</span>, scale_unit <span class="op">=</span> <span class="cn">F</span>, center <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p>Run a UMAP based on PCA dimension reduction and view pre-clustering
UMAP.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_g</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">n_g</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">)</span></span>
<span><span class="va">graph1</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">n_g</span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/giotto-umap-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="clustering">2.3.3 Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h3>
<p>Cells may be clustered into distinct groups based on feature
expression. To cluster, the Giotto Object must contain data that has
undergone PCA dimension reduction, either tSNE or UMAP dimension
reduction, and have a neighbor network.</p>
<p>Create a shared nearest neighbor network (sNN), where k is the number
of k neighbors to use:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">n_g</span>, type <span class="op">=</span> <span class="st">"sNN"</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span>, k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span></code></pre></div>
<p>Cells can be clustered in Giotto using k-means, Leiden, or Louvain
clustering. These clustering algorithms return cluster information
within cell_metadata, which is named accordingly by default. The name
may be changed by providing the name argument, as shown in the code
chunk below.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## k-means clustering</span></span>
<span><span class="va">n_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doKmeans.html">doKmeans</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">n_g</span>, dim_reduction_to_use <span class="op">=</span> <span class="st">'pca'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Quick-TRANSfer stage steps exceeded maximum (= 249600)</span></span></code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Plot UMAP post-clustering to visualize kmeans</span></span>
<span><span class="va">graph2</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span></span>
<span>    gobject <span class="op">=</span> <span class="va">n_g</span>,</span>
<span>    cell_color <span class="op">=</span> <span class="st">'kmeans'</span>,</span>
<span>    show_NN_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">2.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/giotto-kmeans-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="spatial-co-expression">2.3.4 Spatial co-expression<a class="anchor" aria-label="anchor" href="#spatial-co-expression"></a>
</h3>
<p>Using previously created spatial network to calculate spatially
variable genes using binSpect method.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_g</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialDelaunayNetwork.html">createSpatialDelaunayNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">n_g</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoSpatNetworks.html">showGiottoSpatNetworks</a></span><span class="op">(</span><span class="va">n_g</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## └──Spatial unit "<span style="color: #0000BB;">cell</span>"</span></span>
<span><span class="co">##    └──S4 spatialNetworkObj "<span style="color: #00BBBB;">Delaunay_network</span>"   (11694 rows)</span></span>
<span><span class="co">##                           from                 to sdimx_begin sdimy_begin sdimx_end</span></span>
<span><span class="co">##                         &lt;char&gt;             &lt;char&gt;       &lt;num&gt;       &lt;num&gt;     &lt;num&gt;</span></span>
<span><span class="co">##          1: AAACAACGAATAGTTC-1 CCAAGCTTGATCTCCT-1        3913        2435      4050</span></span>
<span><span class="co">##          2: AAACAACGAATAGTTC-1 CAAGGGAGTGTATTTG-1        3913        2435      3981</span></span>
<span><span class="co">##          3: AAACAACGAATAGTTC-1 GGGTTTCCGGCTTCCA-1        3913        2435      3775</span></span>
<span><span class="co">##          4: AAACAACGAATAGTTC-1 TAACCGTCCAGTTCAT-1        3913        2435      3843</span></span>
<span><span class="co">##             sdimy_end distance      weight</span></span>
<span><span class="co">##                 &lt;num&gt;    &lt;num&gt;       &lt;num&gt;</span></span>
<span><span class="co">##          1:      2436 137.0036 0.007299076</span></span>
<span><span class="co">##          2:      2555 137.9275 0.007250185</span></span>
<span><span class="co">##          3:      2434 138.0036 0.007246187</span></span>
<span><span class="co">##          4:      2554 138.0616 0.007243145</span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ranktest</span> <span class="op">=</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span></span>
<span>    <span class="va">n_g</span>, bin_method <span class="op">=</span> <span class="st">'rank'</span>,</span>
<span>    calc_hub <span class="op">=</span> <span class="cn">T</span>, hub_min_int <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  This is the single parameter version of binSpect</span></span>
<span><span class="co">##  1. matrix binarization complete </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  2. spatial enrichment test completed </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  3. (optional) average expression of high expressing cells calculated </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  4. (optional) number of high expressing cells calculated</span></span></code></pre>
<p>Here we will subset on the top 300 spatial genes identified with
binSpect. We will also show how to identify the top spatially correlated
genes.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3.1 cluster the top 500 spatial genes into 20 clusters</span></span>
<span><span class="va">ext_spatial_genes</span> <span class="op">=</span> <span class="va">ranktest</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">300</span>,<span class="op">]</span><span class="op">$</span><span class="va">feats</span></span>
<span></span>
<span><span class="co"># here we use existing detectSpatialCorGenes function to calculate pairwise distances between genes (but set network_smoothing=0 to use default clustering)</span></span>
<span><span class="va">spat_cor_netw_DT</span> <span class="op">=</span> <span class="fu"><a href="../reference/detectSpatialCorFeats.html">detectSpatialCorFeats</a></span><span class="op">(</span></span>
<span>    <span class="va">n_g</span>,</span>
<span>    method <span class="op">=</span> <span class="st">'network'</span>,</span>
<span>    spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>    subset_feats <span class="op">=</span> <span class="va">ext_spatial_genes</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># 3.2 identify most similar spatially correlated genes for one gene</span></span>
<span><span class="va">top10_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/showSpatialCorFeats.html">showSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, feats <span class="op">=</span> <span class="st">'Dsp'</span>, show_top_feats <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>Use the pheatmap to identify spatial co-expression modules and adjust
the number of clusters (k) accordingly.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3.3 identify potenial spatial co-expression</span></span>
<span><span class="va">spat_cor_netw_DT</span> <span class="op">=</span> <span class="fu"><a href="../reference/clusterSpatialCorFeats.html">clusterSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>, k <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize clusters</span></span>
<span><span class="va">graph3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/heatmSpatialCorFeats.html">heatmSpatialCorFeats</a></span><span class="op">(</span></span>
<span>    <span class="va">n_g</span>,</span>
<span>    spatCorObject <span class="op">=</span> <span class="va">spat_cor_netw_DT</span>,</span>
<span>    use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>,</span>
<span>    heatmap_legend_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>title <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>,</span>
<span>    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">6</span>, base_width <span class="op">=</span> <span class="fl">8</span>, units <span class="op">=</span> <span class="st">'cm'</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/coexpression-modules-1.png" width="700"></p>
<p>genes from each co-expression module and then aggregate genes into
meta genes.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3.4 create metagenes / co-expression modules</span></span>
<span><span class="va">cluster_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBalancedSpatCoexpressionFeats.html">getBalancedSpatCoexpressionFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, maximum <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1</span></span>
<span><span class="co">## [1] 2</span></span>
<span><span class="co">## [1] 3</span></span>
<span><span class="co">## [1] 4</span></span>
<span><span class="co">## [1] 5</span></span>
<span><span class="co">## [1] 6</span></span>
<span><span class="co">## [1] 7</span></span></code></pre>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_g</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createMetafeats.html">createMetafeats</a></span><span class="op">(</span><span class="va">n_g</span>, feat_clusters <span class="op">=</span> <span class="va">cluster_genes</span>, name <span class="op">=</span> <span class="st">'cluster_metagene'</span><span class="op">)</span>                     </span>
<span>    </span>
<span>    </span>
<span><span class="va">graph4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatCellPlot.html">spatCellPlot</a></span><span class="op">(</span></span>
<span>    <span class="va">n_g</span>,</span>
<span>    spat_enr_names <span class="op">=</span> <span class="st">'cluster_metagene'</span>,</span>
<span>    cell_annotation_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">1</span>, cow_n_col <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span>   </span></code></pre></div>
<p><img src="interoperability_spatialexperiment_files/figure-html/create-meta-feats-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">3.0 Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] STexampleData_1.10.0        ExperimentHub_2.10.0       </span></span>
<span><span class="co">##  [3] AnnotationHub_3.10.0        BiocFileCache_2.10.1       </span></span>
<span><span class="co">##  [5] dbplyr_2.4.0                scran_1.30.2               </span></span>
<span><span class="co">##  [7] scater_1.30.1               scuttle_1.12.0             </span></span>
<span><span class="co">##  [9] pheatmap_1.0.12             ggspavis_1.8.0             </span></span>
<span><span class="co">## [11] ggplot2_3.4.4               SpatialExperiment_1.12.0   </span></span>
<span><span class="co">## [13] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0</span></span>
<span><span class="co">## [15] Biobase_2.62.0              GenomicRanges_1.54.1       </span></span>
<span><span class="co">## [17] GenomeInfoDb_1.38.5         IRanges_2.36.0             </span></span>
<span><span class="co">## [19] S4Vectors_0.40.2            BiocGenerics_0.48.1        </span></span>
<span><span class="co">## [21] MatrixGenerics_1.14.0       matrixStats_1.2.0          </span></span>
<span><span class="co">## [23] GiottoData_0.2.6.2          GiottoUtils_0.1.3          </span></span>
<span><span class="co">## [25] Giotto_4.0.2                GiottoClass_0.1.3          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RcppAnnoy_0.0.22              later_1.3.2                  </span></span>
<span><span class="co">##   [3] bitops_1.0-7                  filelock_1.0.3               </span></span>
<span><span class="co">##   [5] tibble_3.2.1                  lifecycle_1.0.4              </span></span>
<span><span class="co">##   [7] doParallel_1.0.17             edgeR_4.0.14                 </span></span>
<span><span class="co">##   [9] globals_0.16.2                lattice_0.21-9               </span></span>
<span><span class="co">##  [11] backports_1.4.1               magrittr_2.0.3               </span></span>
<span><span class="co">##  [13] limma_3.58.1                  sass_0.4.8                   </span></span>
<span><span class="co">##  [15] rmarkdown_2.25                jquerylib_0.1.4              </span></span>
<span><span class="co">##  [17] yaml_2.3.8                    metapod_1.10.1               </span></span>
<span><span class="co">##  [19] httpuv_1.6.14                 ggside_0.2.3                 </span></span>
<span><span class="co">##  [21] reticulate_1.35.0             cowplot_1.1.3                </span></span>
<span><span class="co">##  [23] DBI_1.2.1                     RColorBrewer_1.1-3           </span></span>
<span><span class="co">##  [25] abind_1.4-5                   zlibbioc_1.48.0              </span></span>
<span><span class="co">##  [27] purrr_1.0.2                   RCurl_1.98-1.14              </span></span>
<span><span class="co">##  [29] rappdirs_0.3.3                circlize_0.4.15              </span></span>
<span><span class="co">##  [31] GenomeInfoDbData_1.2.11       ggrepel_0.9.5                </span></span>
<span><span class="co">##  [33] irlba_2.3.5.1                 listenv_0.9.1                </span></span>
<span><span class="co">##  [35] terra_1.7-71                  parallelly_1.36.0            </span></span>
<span><span class="co">##  [37] dqrng_0.3.2                   pkgdown_2.0.7                </span></span>
<span><span class="co">##  [39] DelayedMatrixStats_1.24.0     colorRamp2_0.1.0             </span></span>
<span><span class="co">##  [41] codetools_0.2-19              DelayedArray_0.28.0          </span></span>
<span><span class="co">##  [43] shape_1.4.6                   tidyselect_1.2.0             </span></span>
<span><span class="co">##  [45] farver_2.1.1                  ScaledMatrix_1.10.0          </span></span>
<span><span class="co">##  [47] viridis_0.6.5                 jsonlite_1.8.8               </span></span>
<span><span class="co">##  [49] GetoptLong_1.0.5              BiocNeighbors_1.20.2         </span></span>
<span><span class="co">##  [51] ellipsis_0.3.2                iterators_1.0.14             </span></span>
<span><span class="co">##  [53] systemfonts_1.0.5             foreach_1.5.2                </span></span>
<span><span class="co">##  [55] dbscan_1.1-12                 tools_4.3.2                  </span></span>
<span><span class="co">##  [57] ragg_1.2.7                    Rcpp_1.0.12                  </span></span>
<span><span class="co">##  [59] glue_1.7.0                    gridExtra_2.3                </span></span>
<span><span class="co">##  [61] SparseArray_1.2.3             xfun_0.41                    </span></span>
<span><span class="co">##  [63] dplyr_1.1.4                   withr_3.0.0                  </span></span>
<span><span class="co">##  [65] BiocManager_1.30.22           fastmap_1.1.1                </span></span>
<span><span class="co">##  [67] bluster_1.12.0                fansi_1.0.6                  </span></span>
<span><span class="co">##  [69] digest_0.6.34                 rsvd_1.0.5                   </span></span>
<span><span class="co">##  [71] R6_2.5.1                      mime_0.12                    </span></span>
<span><span class="co">##  [73] textshaping_0.3.7             colorspace_2.1-0             </span></span>
<span><span class="co">##  [75] Cairo_1.6-2                   gtools_3.9.5                 </span></span>
<span><span class="co">##  [77] RSQLite_2.3.5                 utf8_1.2.4                   </span></span>
<span><span class="co">##  [79] generics_0.1.3                data.table_1.15.0            </span></span>
<span><span class="co">##  [81] FNN_1.1.4                     httr_1.4.7                   </span></span>
<span><span class="co">##  [83] S4Arrays_1.2.0                uwot_0.1.16                  </span></span>
<span><span class="co">##  [85] pkgconfig_2.0.3               gtable_0.3.4                 </span></span>
<span><span class="co">##  [87] blob_1.2.4                    ComplexHeatmap_2.18.0        </span></span>
<span><span class="co">##  [89] GiottoVisuals_0.1.4           XVector_0.42.0               </span></span>
<span><span class="co">##  [91] htmltools_0.5.7               clue_0.3-65                  </span></span>
<span><span class="co">##  [93] scales_1.3.0                  png_0.1-8                    </span></span>
<span><span class="co">##  [95] knitr_1.45                    rjson_0.2.21                 </span></span>
<span><span class="co">##  [97] checkmate_2.3.1               curl_5.2.0                   </span></span>
<span><span class="co">##  [99] GlobalOptions_0.1.2           cachem_1.0.8                 </span></span>
<span><span class="co">## [101] stringr_1.5.1                 BiocVersion_3.18.1           </span></span>
<span><span class="co">## [103] parallel_4.3.2                vipor_0.4.7                  </span></span>
<span><span class="co">## [105] AnnotationDbi_1.64.1          desc_1.4.3                   </span></span>
<span><span class="co">## [107] pillar_1.9.0                  grid_4.3.2                   </span></span>
<span><span class="co">## [109] vctrs_0.6.5                   promises_1.2.1               </span></span>
<span><span class="co">## [111] BiocSingular_1.18.0           beachmat_2.18.0              </span></span>
<span><span class="co">## [113] xtable_1.8-4                  cluster_2.1.4                </span></span>
<span><span class="co">## [115] beeswarm_0.4.0                evaluate_0.23                </span></span>
<span><span class="co">## [117] magick_2.8.2                  cli_3.6.2                    </span></span>
<span><span class="co">## [119] locfit_1.5-9.8                compiler_4.3.2               </span></span>
<span><span class="co">## [121] rlang_1.1.3                   crayon_1.5.2                 </span></span>
<span><span class="co">## [123] future.apply_1.11.1           labeling_0.4.3               </span></span>
<span><span class="co">## [125] fs_1.6.3                      ggbeeswarm_0.7.2             </span></span>
<span><span class="co">## [127] stringi_1.8.3                 deldir_2.0-2                 </span></span>
<span><span class="co">## [129] viridisLite_0.4.2             BiocParallel_1.36.0          </span></span>
<span><span class="co">## [131] munsell_0.5.0                 Biostrings_2.70.2            </span></span>
<span><span class="co">## [133] Matrix_1.6-5                  sparseMatrixStats_1.14.0     </span></span>
<span><span class="co">## [135] bit64_4.0.5                   future_1.33.1                </span></span>
<span><span class="co">## [137] KEGGREST_1.42.0               statmod_1.5.0                </span></span>
<span><span class="co">## [139] shiny_1.8.0                   interactiveDisplayBase_1.40.0</span></span>
<span><span class="co">## [141] highr_0.10                    igraph_2.0.1.1               </span></span>
<span><span class="co">## [143] memoise_2.0.1                 bslib_0.6.1                  </span></span>
<span><span class="co">## [145] bit_4.0.5</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
