<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interactive selection â€¢ Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Interactive selection">
<meta property="og:description" content="Giotto">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
    <li>
      <a href="../articles/contributing.html">Contributing guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html">Spatial geometries</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Interactive selection</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/HEAD/vignettes/interactive_selection.Rmd" class="external-link"><code>vignettes/interactive_selection.Rmd</code></a></small>
      <div class="hidden name"><code>interactive_selection.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="set-up-giotto-object">Set up Giotto object<a class="anchor" aria-label="anchor" href="#set-up-giotto-object"></a>
</h2>
<p>NOTE: To follow this tutorial, you must pre-run the Parts 1-4 from
the <a href="https://giottosuite.readthedocs.io/en/latest/subsections/datasets/mouse_visium_brain.html" class="external-link">Mouse
Visium Brain tutorial</a></p>
<p>You can use as input either a ggplot2 object, a terra::rast image, or
the output of running the spatPlot2D( ) function.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">'Giotto'</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure GiottoData, a small, helper module for tutorials, is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">'GiottoData'</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">'drieslab/GiottoData'</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed.</span></span>
<span><span class="va">genv_exists</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/checkGiottoEnvironment.html">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/installGiottoEnvironment.html">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-giotto-spatplot-object">1. Create a Giotto spatPlot object<a class="anchor" aria-label="anchor" href="#create-a-giotto-spatplot-object"></a>
</h2>
<p>When creating the spatPlot2D object, you can only plot the spatial
cells/spots or show the tissue image in the background.</p>
<p>If you plan to select individual clusters or cell types, we recommend
coloring the cells/spots using the cell_color argument.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot spatial cells/spots</span></span>
<span><span class="va">my_spatPlot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium_brain</span>,</span>
<span>                          cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>                          point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                          point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                          show_image <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Plot spatial cells/spots with tissue on the background</span></span>
<span><span class="va">my_spatPlot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium_brain</span>,</span>
<span>                          point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                          point_alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                          show_image <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-the-interactive-selection-tool">2. Run the interactive selection tool<a class="anchor" aria-label="anchor" href="#run-the-interactive-selection-tool"></a>
</h2>
<p>plotInteractivePolygons( ) will run a ShinyApp gadget within your
RStudio session. Use your cursor to draw the area of interest and add a
name. You can draw as many regions as you wish.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run the Shiny app with default parameters</span></span>
<span><span class="fu"><a href="../reference/plotInteractivePolygons.html">plotInteractivePolygons</a></span><span class="op">(</span><span class="va">my_spatPlot</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/1-interactive_brain_spots.png"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run the Shiny app using the tissue image</span></span>
<span><span class="fu"><a href="../reference/plotInteractivePolygons.html">plotInteractivePolygons</a></span><span class="op">(</span><span class="va">my_spatPlot</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/2-interactive_brain_tissue.png"></p>
<p>You can use additional parameters passed to the polygon( ) function,
such as the line thickness and color of the polygons.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Modify the width and/or color of the line</span></span>
<span><span class="fu"><a href="../reference/plotInteractivePolygons.html">plotInteractivePolygons</a></span><span class="op">(</span><span class="va">my_spatPlot</span>,</span>
<span>                        color <span class="op">=</span> <span class="st">'black'</span>,</span>
<span>                        lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/3-interactive_brain_black.png"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Modify the width or heigh of the plotted image</span></span>
<span><span class="fu"><a href="../reference/plotInteractivePolygons.html">plotInteractivePolygons</a></span><span class="op">(</span><span class="va">my_spatPlot</span>,</span>
<span>                        height <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/4-interactive_brain_height.png"></p>
<p>Use the slider bars to zoom in or out your plot and draw tiny
polygons.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Use the zoom in and out slide-bars to select small areas</span></span>
<span><span class="fu"><a href="../reference/plotInteractivePolygons.html">plotInteractivePolygons</a></span><span class="op">(</span><span class="va">my_spatPlot</span>,</span>
<span>                        height <span class="op">=</span> <span class="fl">320</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/5-interactive_brain_zoom.png"></p>
</div>
<div class="section level2">
<h2 id="save-the-coordinates">3. Save the coordinates<a class="anchor" aria-label="anchor" href="#save-the-coordinates"></a>
</h2>
<p>Click on the button <em>Done</em> once you have finished selecting
the areas of interest.</p>
<p>By default, plotInteractivePolygons() will print the polygon x and y
coordinates as a â€˜data.tableâ€™ object on the Console panel, but you can
store the output using the assignment operator.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The polygon coordinates will be saved on my_polygon_coordinates as a data.table object</span></span>
<span><span class="va">my_polygon_coordinates</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotInteractivePolygons.html">plotInteractivePolygons</a></span><span class="op">(</span><span class="va">my_spatPlot</span>,</span>
<span>                                                  height <span class="op">=</span> <span class="fl">320</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/6-my_polygon_coordinates.png"></p>
<p>Now, letâ€™s add the polygon coordinates to the Giotto object:</p>
<p>Create a polygon object</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We must transform the data.table or data.frame with coordinates into a Giotto polygon object</span></span>
<span><span class="va">my_giotto_polygons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPolygon.html">createGiottoPolygonsFromDfr</a></span><span class="op">(</span><span class="va">my_polygon_coordinates</span>, </span>
<span>                                                  name <span class="op">=</span> <span class="st">'selections'</span><span class="op">)</span></span></code></pre></div>
<p>If you plan to save and load your Giotto object after adding the
selected polygons, include the centroids of the polygon by using the
following argument:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We must transform the data.table or data.frame with coordinates into a Giotto polygon object</span></span>
<span><span class="va">my_giotto_polygons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPolygon.html">createGiottoPolygonsFromDfr</a></span><span class="op">(</span><span class="va">my_polygon_coordinates</span>, </span>
<span>                                                  name <span class="op">=</span> <span class="st">'selections'</span>,</span>
<span>                                                  calc_centroids <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Then, add the polygons to the Giotto object</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">visium_brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addGiottoPolygons.html">addGiottoPolygons</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium_brain</span>,</span>
<span>                                  gpolygons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">my_giotto_polygons</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Finally, add the corresponding polygon ID to the cell Metadata. By
default, the function will add a â€˜no_polygonâ€™ label to cells outside
polygons, but you can customize it using the na.label argument.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Using the default parameters</span></span>
<span><span class="va">visium_brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addPolygonCells.html">addPolygonCells</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>                                polygon_name <span class="op">=</span> <span class="st">'selections'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Let's see how it looks like now the cell_metadata</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">visium_brain</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/7-new_metadata.png"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Customize the NA label</span></span>
<span><span class="va">visium_brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addPolygonCells.html">addPolygonCells</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>                                polygon_name <span class="op">=</span> <span class="st">'selections'</span>,</span>
<span>                                na.label <span class="op">=</span> <span class="st">'No Polygon'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/8-new_metadata_customized.png"></p>
</div>
<div class="section level2">
<h2 id="get-the-cells-located-within-each-polygon">4. Get the cells located within each polygon<a class="anchor" aria-label="anchor" href="#get-the-cells-located-within-each-polygon"></a>
</h2>
<p>You can extract the coordinates and IDs from cells located within one
or multiple regions of interest.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Provide the name of the polygon to extract cells from</span></span>
<span><span class="fu"><a href="../reference/getCellsFromPolygon.html">getCellsFromPolygon</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>                    polygon_name <span class="op">=</span> <span class="st">'selections'</span>,</span>
<span>                    polygons <span class="op">=</span> <span class="st">'polygon 1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/9-get_cells_polygon_1.png"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## If no polygon name is provided, the function will retrieve cells located within all polygons</span></span>
<span><span class="fu"><a href="../reference/getCellsFromPolygon.html">getCellsFromPolygon</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>                    polygon_name <span class="op">=</span> <span class="st">'selections'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/10-get_cells.png"></p>
</div>
<div class="section level2">
<h2 id="compare-gene-expression">5. Compare gene expression<a class="anchor" aria-label="anchor" href="#compare-gene-expression"></a>
</h2>
<p>Letâ€™s compare the expression levels of some genes of interest between
the selected regions.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## You can provide a list of genes</span></span>
<span><span class="fu"><a href="../reference/comparePolygonExpression.html">comparePolygonExpression</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>                         selected_feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Stmn1'</span>, <span class="st">'Psd'</span>, <span class="st">'Ly6h'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/11-compare_genes.png"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Or calculate the top genes expressed within each region, then provide the result to compare polygons.</span></span>
<span><span class="va">scran_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>                                        spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                                        feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>                                        method <span class="op">=</span> <span class="st">'scran'</span>,</span>
<span>                                        expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                                        cluster_column <span class="op">=</span> <span class="st">'selections'</span>,</span>
<span>                                        min_feats <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">top_genes</span> <span class="op">&lt;-</span> <span class="va">scran_results</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span></span>
<span></span>
<span><span class="fu"><a href="../reference/comparePolygonExpression.html">comparePolygonExpression</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>                         selected_feats <span class="op">=</span> <span class="va">top_genes</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/12-compare_topgenes.png"></p>
</div>
<div class="section level2">
<h2 id="compare-cell-type-abundance">6. Compare cell type abundance<a class="anchor" aria-label="anchor" href="#compare-cell-type-abundance"></a>
</h2>
<p>If you have run an analysis for finding clusters or have annotated
cell types within the cell metadata slot, you can use the column name to
compare the abundance of cell types between the selected regions.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## By default, the function will look for the leiden_cluster</span></span>
<span><span class="fu"><a href="../reference/compareCellAbundance.html">compareCellAbundance</a></span><span class="op">(</span><span class="va">visium_brain</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## You can use other columns within the cell metadata table to compare the cell type abundances</span></span>
<span><span class="fu"><a href="../reference/compareCellAbundance.html">compareCellAbundance</a></span><span class="op">(</span><span class="va">visium_brain</span>, cell_type_column <span class="op">=</span> <span class="st">'cell_type'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/13-compare_cell_abundance.png"></p>
</div>
<div class="section level2">
<h2 id="create-a-spatplot-per-region">7. Create a spatPlot per region<a class="anchor" aria-label="anchor" href="#create-a-spatplot-per-region"></a>
</h2>
<p>You can use the spatPlot( ) arguments to isolate and plot each
region. Also, you can color each cell by cluster, cell type or
expression level.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compare clusters within each region</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot2D</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">'selections'</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>           coord_fix_ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>           cow_n_col <span class="op">=</span> <span class="fl">3</span>,</span>
<span>           show_legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/14-compare_spatplots.png"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compare expression levels between regions</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatFeatPlot2D.html">spatFeatPlot2D</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>               expression_values <span class="op">=</span> <span class="st">'scaled'</span>,</span>
<span>               group_by <span class="op">=</span> <span class="st">'selections'</span>,</span>
<span>               feats <span class="op">=</span> <span class="st">'Psd'</span>,</span>
<span>               point_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/15-compare_spatfeatplot.png"></p>
</div>
<div class="section level2">
<h2 id="plot-again-the-polygons">8. Plot again the polygons<a class="anchor" aria-label="anchor" href="#plot-again-the-polygons"></a>
</h2>
<p>You can plot again all or some drawn polygons. The minimal input is
the Giotto object containing polygon coordinates within the spatial_info
slot, as well as the spatial plot.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot all polygons</span></span>
<span><span class="fu"><a href="../reference/plotPolygons.html">plotPolygons</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>             polygon_name <span class="op">=</span> <span class="st">'selections'</span>,</span>
<span>             x <span class="op">=</span> <span class="va">my_spatPlot</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/16-plotPolygons.png"></p>
<p>Additionaly, you can plot a few polygons by providing a vector with
the polygon IDs.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot a subset of polygons</span></span>
<span><span class="fu"><a href="../reference/plotPolygons.html">plotPolygons</a></span><span class="op">(</span><span class="va">visium_brain</span>,</span>
<span>             polygon_name <span class="op">=</span> <span class="st">'selections'</span>,</span>
<span>             x <span class="op">=</span> <span class="va">my_spatPlot</span>,</span>
<span>             polygons <span class="op">=</span> <span class="st">'polygon 2'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interactive_selection/17-plotPolygon2.png"></p>
</div>
<div class="section level2">
<h2 id="session-info">9. Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">3.2</span> (<span class="dv">2023-10-31</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>apple<span class="sc">-</span><span class="fu">darwin20</span> (<span class="dv">64</span><span class="sc">-</span>bit)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="dv">14</span>.<span class="fl">3.1</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.3</span><span class="sc">-</span>x86_64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] shiny_1.<span class="fl">8.0</span>       Giotto_4.<span class="fl">0.2</span>      GiottoClass_0.<span class="fl">1.3</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">1</span>] colorRamp2_0.<span class="fl">1.0</span>            bitops_1<span class="fl">.0</span><span class="dv">-7</span>                rlang_1.<span class="fl">1.3</span>                </span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">4</span>] magrittr_2.<span class="fl">0.3</span>              GiottoUtils_0.<span class="fl">1.5</span>           matrixStats_1.<span class="fl">2.0</span>          </span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">7</span>] compiler_4.<span class="fl">3.2</span>              png_0<span class="fl">.1</span><span class="dv">-8</span>                   systemfonts_1.<span class="fl">0.5</span>          </span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a> [<span class="dv">10</span>] vctrs_0.<span class="fl">6.5</span>                 pkgconfig_2.<span class="fl">0.3</span>             SpatialExperiment_1.<span class="fl">12.0</span>   </span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a> [<span class="dv">13</span>] crayon_1.<span class="fl">5.2</span>                fastmap_1.<span class="fl">1.1</span>               ellipsis_0.<span class="fl">3.2</span>             </span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a> [<span class="dv">16</span>] backports_1.<span class="fl">4.1</span>             magick_2.<span class="fl">8.2</span>                XVector_0.<span class="fl">42.0</span>             </span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a> [<span class="dv">19</span>] labeling_0.<span class="fl">4.3</span>              utf8_1.<span class="fl">2.4</span>                  promises_1.<span class="fl">2.1</span>             </span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a> [<span class="dv">22</span>] rmarkdown_2<span class="fl">.25</span>              ragg_1.<span class="fl">2.7</span>                  xfun_0<span class="fl">.42</span>                  </span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a> [<span class="dv">25</span>] cachem_1.<span class="fl">0.8</span>                zlibbioc_1.<span class="fl">48.0</span>             beachmat_2.<span class="fl">18.0</span>            </span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a> [<span class="dv">28</span>] GenomeInfoDb_1.<span class="fl">38.6</span>         jsonlite_1.<span class="fl">8.8</span>              later_1.<span class="fl">3.2</span>                </span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a> [<span class="dv">31</span>] DelayedArray_0.<span class="fl">28.0</span>         BiocParallel_1.<span class="fl">36.0</span>         terra_1<span class="fl">.7</span><span class="dv">-71</span>               </span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a> [<span class="dv">34</span>] irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>               parallel_4.<span class="fl">3.2</span>              R6_2.<span class="fl">5.1</span>                   </span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a> [<span class="dv">37</span>] bslib_0.<span class="fl">6.1</span>                 RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>          reticulate_1.<span class="fl">35.0</span>          </span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a> [<span class="dv">40</span>] parallelly_1.<span class="fl">36.0</span>           jquerylib_0.<span class="fl">1.4</span>             GenomicRanges_1.<span class="fl">54.1</span>       </span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a> [<span class="dv">43</span>] Rcpp_1.<span class="fl">0.12</span>                 SummarizedExperiment_1.<span class="fl">32.0</span> knitr_1<span class="fl">.45</span>                 </span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a> [<span class="dv">46</span>] future.apply_1.<span class="fl">11.1</span>         R.utils_2.<span class="fl">12.3</span>              IRanges_2.<span class="fl">36.0</span>             </span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a> [<span class="dv">49</span>] FNN_1.<span class="fl">1.4</span>                   httpuv_1.<span class="fl">6.14</span>               Matrix_1<span class="fl">.6</span><span class="dv">-5</span>               </span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a> [<span class="dv">52</span>] igraph_2.<span class="dv">0</span>.<span class="fl">1.1</span>              tidyselect_1.<span class="fl">2.0</span>            rstudioapi_0.<span class="fl">15.0</span>          </span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a> [<span class="dv">55</span>] abind_1<span class="fl">.4</span><span class="dv">-5</span>                 yaml_2.<span class="fl">3.8</span>                  codetools_0<span class="fl">.2</span><span class="dv">-19</span>           </span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a> [<span class="dv">58</span>] miniUI_0.<span class="dv">1</span>.<span class="fl">1.1</span>              listenv_0.<span class="fl">9.1</span>               lattice_0<span class="fl">.22</span><span class="dv">-5</span>             </span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a> [<span class="dv">61</span>] tibble_3.<span class="fl">2.1</span>                Biobase_2.<span class="fl">62.0</span>              withr_3.<span class="fl">0.0</span>                </span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a> [<span class="dv">64</span>] evaluate_0<span class="fl">.23</span>               Rtsne_0<span class="fl">.17</span>                  future_1.<span class="fl">33.1</span>              </span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a> [<span class="dv">67</span>] pillar_1.<span class="fl">9.0</span>                MatrixGenerics_1.<span class="fl">14.0</span>       checkmate_2.<span class="fl">3.1</span>            </span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a> [<span class="dv">70</span>] stats4_4.<span class="fl">3.2</span>                generics_0.<span class="fl">1.3</span>              dbscan_1<span class="fl">.1</span><span class="dv">-12</span>              </span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a> [<span class="dv">73</span>] RCurl_1<span class="fl">.98-1.14</span>             S4Vectors_0.<span class="fl">40.2</span>            ggplot2_3.<span class="fl">4.4</span>              </span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a> [<span class="dv">76</span>] munsell_0.<span class="fl">5.0</span>               scales_1.<span class="fl">3.0</span>                xtable_1<span class="fl">.8</span><span class="dv">-4</span>               </span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a> [<span class="dv">79</span>] gtools_3.<span class="fl">9.5</span>                globals_0.<span class="fl">16.2</span>              glue_1.<span class="fl">7.0</span>                 </span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a> [<span class="dv">82</span>] tools_4.<span class="fl">3.2</span>                 GiottoVisuals_0.<span class="fl">1.4</span>         data.table_1.<span class="fl">15.0</span>          </span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a> [<span class="dv">85</span>] ScaledMatrix_1.<span class="fl">10.0</span>         cowplot_1.<span class="fl">1.3</span>               grid_4.<span class="fl">3.2</span>                 </span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a> [<span class="dv">88</span>] colorspace_2<span class="fl">.1</span><span class="dv">-0</span>            SingleCellExperiment_1.<span class="fl">24.0</span> GenomeInfoDbData_1.<span class="fl">2.11</span>    </span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a> [<span class="dv">91</span>] BiocSingular_1.<span class="fl">18.0</span>         cli_3.<span class="fl">6.2</span>                   rsvd_1.<span class="fl">0.5</span>                 </span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a> [<span class="dv">94</span>] textshaping_0.<span class="fl">3.7</span>           fansi_1.<span class="fl">0.6</span>                 S4Arrays_1.<span class="fl">2.0</span>             </span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a> [<span class="dv">97</span>] dplyr_1.<span class="fl">1.4</span>                 uwot_0.<span class="fl">1.16</span>                 gtable_0.<span class="fl">3.4</span>               </span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>[<span class="dv">100</span>] R.methodsS3_1.<span class="fl">8.2</span>           sass_0.<span class="fl">4.8</span>                  digest_0.<span class="fl">6.34</span>              </span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>[<span class="dv">103</span>] BiocGenerics_0.<span class="fl">48.1</span>         SparseArray_1.<span class="fl">2.3</span>           ggrepel_0.<span class="fl">9.5</span>              </span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a>[<span class="dv">106</span>] rjson_0.<span class="fl">2.21</span>                farver_2.<span class="fl">1.1</span>                memoise_2.<span class="fl">0.1</span>              </span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a>[<span class="dv">109</span>] htmltools_0.<span class="fl">5.7</span>             R.oo_1.<span class="fl">26.0</span>                 lifecycle_1.<span class="fl">0.4</span>            </span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a>[<span class="dv">112</span>] mime_0<span class="fl">.12</span>   </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. ChÃ¡vez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew Oâ€™Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
