<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>mini seqFISH â€¢ Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="mini seqFISH">
<meta property="og:description" content="Giotto">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/errorsfaqsandtips.html">Solutions to Installation Errors</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html">Spatial geometries</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>mini seqFISH</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/HEAD/vignettes/mini_seqfish.Rmd" class="external-link"><code>vignettes/mini_seqfish.Rmd</code></a></small>
      <div class="hidden name"><code>mini_seqfish.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="start-giotto">Start Giotto<a class="anchor" aria-label="anchor" href="#start-giotto"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure GiottoData, a small, helper module for tutorials, is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"GiottoData"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"drieslab/GiottoData"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed.</span></span>
<span><span class="va">genv_exists</span> <span class="op">=</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/checkGiottoEnvironment.html">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/installGiottoEnvironment.html">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-the-giotto-object">1. Load the Giotto object<a class="anchor" aria-label="anchor" href="#load-the-giotto-object"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/">Giotto</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/GiottoData/">GiottoData</a></span><span class="op">)</span></span></code></pre></div>
<p>A small seqFISH data is available through the giottoData package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download data</span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/loadGiottoMini.html">loadGiottoMini</a></span><span class="op">(</span><span class="st">"seqfish"</span>,</span>
<span>                              python_path <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="set-giotto-instructions-optional">Set Giotto instructions (optional)<a class="anchor" aria-label="anchor" href="#set-giotto-instructions-optional"></a>
</h3>
<p>How to work with Giotto instructions that are part of your Giotto
object:</p>
<ul>
<li>Show the instructions associated with your Giotto object with
<strong>showGiottoInstructions()</strong>
</li>
<li>Change one or more instructions with
<strong>changeGiottoInstructions()</strong>
</li>
<li>Replace all instructions at once with
<strong>replaceGiottoInstructions()</strong>
</li>
<li>Read or get a specific Giotto instruction with
<strong>readGiottoInstructions()</strong>
</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show instructions associated with giotto object (seqfish_mini)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoInstructions.html">showGiottoInstructions</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Change one or more instructions</span></span>
<span><span class="co"># to automatically save figures in save_dir, set save_plot to TRUE</span></span>
<span><span class="va">temp_dir</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/changeGiottoInstructions.html">changeGiottoInstructions</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,</span>
<span>                                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"save_dir"</span>, <span class="st">"save_plot"</span>, <span class="st">"show_plot"</span><span class="op">)</span>,</span>
<span>                                          new_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">temp_dir</span>,<span class="cn">TRUE</span>, <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="processing-steps">2. Processing steps<a class="anchor" aria-label="anchor" href="#processing-steps"></a>
</h2>
<ul>
<li>Filter genes and cells based on detection frequencies.</li>
<li>Normalize expression matrix (log transformation, scaling factor
and/or z-scores)</li>
<li>Add cell and gene statistics (optional)</li>
<li>Adjust expression matrix for technical covariates or batches
(optional). These results will be stored in the <em>custom</em>
slot.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                            expression_threshold <span class="op">=</span> <span class="fl">0.5</span>, </span>
<span>                            feat_det_in_min_cells <span class="op">=</span> <span class="fl">20</span>, </span>
<span>                            min_det_feats_per_cell <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                               scalefactor <span class="op">=</span> <span class="fl">6000</span>, </span>
<span>                               verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/adjustGiottoMatrix.html">adjustGiottoMatrix</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                                  expression_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'normalized'</span><span class="op">)</span>, </span>
<span>                                  covariate_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'nr_feats'</span>,</span>
<span>                                                        <span class="st">'total_expr'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dimension-reduction">3. Dimension reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<ul>
<li>Identify highly variable features (HVF)</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/0-HVFplot.png"></p>
<ul>
<li>Perform PCA</li>
<li>Identify number of significant principal components (PCs)</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, ncp <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/1-screePlot.png"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotPCA.html">plotPCA</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/2-PCA.png"></p>
<ul>
<li>Run UMAP and/or t-SNE on PCs (or directly on matrix)</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>                       dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, </span>
<span>                       n_threads <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/3-UMAP.png"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/runtSNE.html">runtSNE</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>                       dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotTSNE.html">plotTSNE</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/4-tSNE.png"></p>
</div>
<div class="section level2">
<h2 id="clustering">4. Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h2>
<ul>
<li>Create a shared (default) nearest network in PCA space (or directly
on matrix)</li>
<li>Cluster on nearest network with Leiden or Louvain (k-means and
hclust are alternatives)</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                                    dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, </span>
<span>                                    k <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                               resolution <span class="op">=</span> <span class="fl">0.4</span>, </span>
<span>                               n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize UMAP cluster results</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>, </span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">T</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/5-UMAP.png"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># visualize UMAP and spatial results</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html">spatDimPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>, </span>
<span>            spat_point_shape <span class="op">=</span> <span class="st">'voronoi'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/6-spatDimPlot2D.png"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># heatmap and dendrogram</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/showClusterHeatmap.html">showClusterHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                   cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/7-showClusterHeatmap.png"></p>
<p>The following step requires the installation of {ggdendro}.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages('ggdendro')</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/andrie/ggdendro" class="external-link">ggdendro</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/showClusterDendrogram.html">showClusterDendrogram</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>                      h <span class="op">=</span> <span class="fl">0.5</span>, </span>
<span>                      rotate <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                      cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/8-showClusterDendrogram.png"></p>
</div>
<div class="section level2">
<h2 id="differential-expression">5. Differential expression<a class="anchor" aria-label="anchor" href="#differential-expression"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gini_markers</span> <span class="op">=</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                                      method <span class="op">=</span> <span class="st">'gini'</span>,</span>
<span>                                      expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                                      cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>                                      min_feats <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                                      min_expr_gini_score <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                                      min_det_gini_score <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get top 2 genes per cluster and visualize with violin plot</span></span>
<span><span class="va">topgenes_gini</span> <span class="op">=</span> <span class="va">gini_markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/violinPlot.html">violinPlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>           feats <span class="op">=</span> <span class="va">topgenes_gini</span><span class="op">$</span><span class="va">feats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>, </span>
<span>           cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/9-violinPlot.png"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get top 6 genes per cluster and visualize with heatmap</span></span>
<span><span class="va">topgenes_gini2</span> <span class="op">=</span> <span class="va">gini_markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">6</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotMetaDataHeatmap.html">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>                    selected_feats <span class="op">=</span> <span class="va">topgenes_gini2</span><span class="op">$</span><span class="va">feats</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'leiden_clus'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/10-plotMetaDataHeatmap.png"></p>
</div>
<div class="section level2">
<h2 id="cell-type-annotation">6. Cell type annotation<a class="anchor" aria-label="anchor" href="#cell-type-annotation"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clusters_cell_types</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cell A'</span>, <span class="st">'cell B'</span>, <span class="st">'cell C'</span>, <span class="st">'cell D'</span>,</span>
<span>                        <span class="st">'cell E'</span>, <span class="st">'cell F'</span>, <span class="st">'cell G'</span>, <span class="st">'cell H'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clusters_cell_types</span><span class="op">)</span> <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span></span>
<span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/annotateGiotto.html">annotateGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                              annotation_vector <span class="op">=</span> <span class="va">clusters_cell_types</span>, </span>
<span>                              cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span>, </span>
<span>                              name <span class="op">=</span> <span class="st">'cell_types'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check new cell metadata</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize annotations</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html">spatDimPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, </span>
<span>            spat_point_size <span class="op">=</span> <span class="fl">3</span>, </span>
<span>            dim_point_size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/11-spatDimPlot2D.png"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># heatmap</span></span>
<span><span class="va">topgenes_heatmap</span> <span class="op">=</span> <span class="va">gini_markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">4</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotHeatmap.html">plotHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>            feats <span class="op">=</span> <span class="va">topgenes_heatmap</span><span class="op">$</span><span class="va">feats</span>, </span>
<span>            feat_order <span class="op">=</span> <span class="st">'custom'</span>,</span>
<span>            feat_custom_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">topgenes_heatmap</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span>,</span>
<span>            cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>            legend_nrows <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/12-plotHeatmap.png"></p>
</div>
<div class="section level2">
<h2 id="spatial-grid">7. Spatial grid<a class="anchor" aria-label="anchor" href="#spatial-grid"></a>
</h2>
<ul>
<li>Create a grid based on defined step sizes in the x,y(,z) axes.</li>
</ul>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialGrid.html">createSpatialGrid</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                                 sdimx_stepsize <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                                 sdimy_stepsize <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                                 minimum_padding <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoSpatGrids.html">showGiottoSpatGrids</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize grid</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>         show_grid <span class="op">=</span> <span class="cn">T</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/13-spatPlot2D.png"></p>
</div>
<div class="section level2">
<h2 id="spatial-network">8. Spatial network<a class="anchor" aria-label="anchor" href="#spatial-network"></a>
</h2>
<ul>
<li>Visualize information about the default Delaunay network</li>
<li>Create a spatial Delaunay network (default)</li>
<li>Create a spatial kNN network</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotStatDelaunayNetwork.html">plotStatDelaunayNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                        maximum_distance <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/14-plotStatDelaunayNetwork.png"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                                    minimum_k <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                                    maximum_distance_delaunay <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                                    minimum_k <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                                    method <span class="op">=</span> <span class="st">'kNN'</span>, k <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoSpatNetworks.html">showGiottoSpatNetworks</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize the two different spatial networks  </span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>         show_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">'blue'</span>, </span>
<span>         spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/15-spatPlot2D.png"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>         show_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">'blue'</span>, </span>
<span>         spatial_network_name <span class="op">=</span> <span class="st">'kNN_network'</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/16-spatPlot2D.png"></p>
</div>
<div class="section level2">
<h2 id="spatial-genes">9. Spatial genes<a class="anchor" aria-label="anchor" href="#spatial-genes"></a>
</h2>
<p>Identify spatial genes with 3 different methods:</p>
<ul>
<li>binSpect with k-means binarization (default)</li>
<li>binSpect with rank binarization</li>
<li>silhouetteRank</li>
</ul>
<p>Visualize top 4 genes per method.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">km_spatialgenes</span> <span class="op">=</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatFeatPlot2D.html">spatFeatPlot2D</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,</span>
<span>               expression_values <span class="op">=</span> <span class="st">'scaled'</span>, </span>
<span>               feats <span class="op">=</span> <span class="va">km_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>,</span>
<span>               point_shape <span class="op">=</span> <span class="st">'border'</span>, </span>
<span>               point_border_stroke <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>               show_network <span class="op">=</span> <span class="cn">F</span>, </span>
<span>               network_color <span class="op">=</span> <span class="st">'lightgrey'</span>, </span>
<span>               point_size <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>               cow_n_col <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/17-spatFeatPlot2D.png"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rank_spatialgenes</span> <span class="op">=</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>                             bin_method <span class="op">=</span> <span class="st">'rank'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatFeatPlot2D.html">spatFeatPlot2D</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>               expression_values <span class="op">=</span> <span class="st">'scaled'</span>, </span>
<span>               feats <span class="op">=</span> <span class="va">rank_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>,</span>
<span>               point_shape <span class="op">=</span> <span class="st">'border'</span>, </span>
<span>               point_border_stroke <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>               show_network <span class="op">=</span> <span class="cn">F</span>, </span>
<span>               network_color <span class="op">=</span> <span class="st">'lightgrey'</span>, </span>
<span>               point_size <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>               cow_n_col <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/18-spatFeatPlot2D.png"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">silh_spatialgenes</span> <span class="op">=</span> <span class="fu"><a href="../reference/silhouetteRank.html">silhouetteRank</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span> <span class="co"># TODO: suppress print output</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatFeatPlot2D.html">spatFeatPlot2D</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>               expression_values <span class="op">=</span> <span class="st">'scaled'</span>, </span>
<span>               feats <span class="op">=</span> <span class="va">silh_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">genes</span>,</span>
<span>               point_shape <span class="op">=</span> <span class="st">'border'</span>, </span>
<span>               point_border_stroke <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>               show_network <span class="op">=</span> <span class="cn">F</span>, </span>
<span>               network_color <span class="op">=</span> <span class="st">'lightgrey'</span>, </span>
<span>               point_size <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>               cow_n_col <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/19-spatFeatPlot2D.png"></p>
</div>
<div class="section level2">
<h2 id="spatial-co-expression-patterns">10. Spatial co-expression patterns<a class="anchor" aria-label="anchor" href="#spatial-co-expression-patterns"></a>
</h2>
<p>Identify robust spatial co-expression patterns using the spatial
network or grid and a subset of individual spatial genes.</p>
<ol style="list-style-type: decimal">
<li>Calculate spatial correlation scores</li>
<li>Cluster correlation scores</li>
</ol>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. calculate spatial correlation scores </span></span>
<span><span class="va">ext_spatial_genes</span> <span class="op">=</span> <span class="va">km_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span></span>
<span></span>
<span><span class="va">spat_cor_netw_DT</span> <span class="op">=</span> <span class="fu"><a href="../reference/detectSpatialCorFeats.html">detectSpatialCorFeats</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,</span>
<span>                                         method <span class="op">=</span> <span class="st">'network'</span>, </span>
<span>                                         spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>                                         subset_feats <span class="op">=</span> <span class="va">ext_spatial_genes</span><span class="op">)</span></span>
<span><span class="co"># 2. cluster correlation scores</span></span>
<span><span class="va">spat_cor_netw_DT</span> <span class="op">=</span> <span class="fu"><a href="../reference/clusterSpatialCorFeats.html">clusterSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, </span>
<span>                                          name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>, </span>
<span>                                          k <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/heatmSpatialCorFeats.html">heatmSpatialCorFeats</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>                     spatCorObject <span class="op">=</span> <span class="va">spat_cor_netw_DT</span>, </span>
<span>                     use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/20-heatmSpatialCorFeats.png"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">netw_ranks</span> <span class="op">=</span> <span class="fu"><a href="../reference/rankSpatialCorGroups.html">rankSpatialCorGroups</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>                                  spatCorObject <span class="op">=</span> <span class="va">spat_cor_netw_DT</span>, </span>
<span>                                  use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/21-rankSpatialCorGroups.png"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_netw_spat_cluster</span> <span class="op">=</span> <span class="fu"><a href="../reference/showSpatialCorFeats.html">showSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, </span>
<span>                                            use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>,</span>
<span>                                            selected_clusters <span class="op">=</span> <span class="fl">6</span>, </span>
<span>                                            show_top_feats <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_genes_DT</span> <span class="op">=</span> <span class="fu"><a href="../reference/showSpatialCorFeats.html">showSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, </span>
<span>                                       use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>,</span>
<span>                                       show_top_feats <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_genes</span> <span class="op">=</span> <span class="va">cluster_genes_DT</span><span class="op">$</span><span class="va">clus</span>; <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cluster_genes</span><span class="op">)</span> <span class="op">=</span> <span class="va">cluster_genes_DT</span><span class="op">$</span><span class="va">feat_ID</span></span>
<span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createMetafeats.html">createMetafeats</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,</span>
<span>                               feat_clusters <span class="op">=</span> <span class="va">cluster_genes</span>,</span>
<span>                               name <span class="op">=</span> <span class="st">'cluster_metagene'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatCellPlot.html">spatCellPlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,</span>
<span>             spat_enr_names <span class="op">=</span> <span class="st">'cluster_metagene'</span>,</span>
<span>             cell_annotation_values <span class="op">=</span> <span class="va">netw_ranks</span><span class="op">$</span><span class="va">clusters</span>,</span>
<span>             point_size <span class="op">=</span> <span class="fl">1.5</span>, cow_n_col <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/22-spatCellPlot2D.png"></p>
</div>
<div class="section level2">
<h2 id="spatial-hmrf-domains">11. Spatial HMRF domains<a class="anchor" aria-label="anchor" href="#spatial-hmrf-domains"></a>
</h2>
<p>The following HMRF function requires {smfishHmrf} .</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remotes::install_bitbucket(repo = 'qzhudfci/smfishhmrf-r', ref='master')</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bitbucket.org/qzhudfci/smfishhmrf-r/src/master/" class="external-link">smfishHmrf</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">hmrf_folder</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">temp_dir</span>,<span class="st">'/'</span>,<span class="st">'11_HMRF/'</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">hmrf_folder</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">hmrf_folder</span>, recursive <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform hmrf</span></span>
<span><span class="va">my_spatial_genes</span> <span class="op">=</span> <span class="va">km_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span></span>
<span></span>
<span><span class="va">HMRF_spatial_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/doHMRF.html">doHMRF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                            expression_values <span class="op">=</span> <span class="st">'scaled'</span>,</span>
<span>                            spatial_genes <span class="op">=</span> <span class="va">my_spatial_genes</span>,</span>
<span>                            spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>                            k <span class="op">=</span> <span class="fl">9</span>,</span>
<span>                            betas <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">28</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>,</span>
<span>                            output_folder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">hmrf_folder</span>, <span class="st">'/'</span>, <span class="st">'Spatial_genes/SG_top100_k9_scaled'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check and select hmrf</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">28</span>, <span class="fl">30</span>, by <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/viewHMRFresults2D.html">viewHMRFresults2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                    HMRFoutput <span class="op">=</span> <span class="va">HMRF_spatial_genes</span>,</span>
<span>                    k <span class="op">=</span> <span class="fl">9</span>, betas_to_view <span class="op">=</span> <span class="va">i</span>,</span>
<span>                    point_size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/addHMRF.html">addHMRF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                       HMRFoutput <span class="op">=</span> <span class="va">HMRF_spatial_genes</span>,</span>
<span>                       k <span class="op">=</span> <span class="fl">9</span>, betas_to_add <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">28</span><span class="op">)</span>,</span>
<span>                       hmrf_name <span class="op">=</span> <span class="st">'HMRF'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize selected hmrf result</span></span>
<span><span class="va">giotto_colors</span> <span class="op">=</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/Giotto/reference/getDistinctColors.html">getDistinctColors</a></span><span class="op">(</span><span class="fl">9</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">giotto_colors</span><span class="op">)</span> <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">9</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">'HMRF_k9_b.28'</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">3</span>, </span>
<span>         coord_fix_ratio <span class="op">=</span> <span class="fl">1</span>, </span>
<span>         cell_color_code <span class="op">=</span> <span class="va">giotto_colors</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/23-spatPlot2D.png"></p>
</div>
<div class="section level2">
<h2 id="cell-neighborhood-cell-typecell-type-interactions">12. Cell neighborhood: cell-type/cell-type interactions<a class="anchor" aria-label="anchor" href="#cell-neighborhood-cell-typecell-type-interactions"></a>
</h2>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">2841</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cell_proximities</span> <span class="op">=</span> <span class="fu"><a href="../reference/cellProximityEnrichment.html">cellProximityEnrichment</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                                           cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                                           spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>                                           adjust_method <span class="op">=</span> <span class="st">'fdr'</span>,</span>
<span>                                           number_of_simulations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># barplot</span></span>
<span><span class="fu"><a href="../reference/cellProximityBarplot.html">cellProximityBarplot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>, </span>
<span>                     min_orig_ints <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                     min_sim_ints <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                     p_val <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/24-cellProximityBarplot.png"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## heatmap</span></span>
<span><span class="fu"><a href="../reference/cellProximityHeatmap.html">cellProximityHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>, </span>
<span>                     order_cell_types <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                     scale <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                     color_breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1.5</span>, <span class="fl">0</span>, <span class="fl">1.5</span><span class="op">)</span>, </span>
<span>                     color_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'blue'</span>, <span class="st">'white'</span>, <span class="st">'red'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/25-cellProximityHeatmap.png"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># network</span></span>
<span><span class="fu"><a href="../reference/cellProximityNetwork.html">cellProximityNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>, </span>
<span>                     remove_self_edges <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                     only_show_enrichment_edges <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/26-cellProximityNetwork.png"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># network with self-edges</span></span>
<span><span class="fu"><a href="../reference/cellProximityNetwork.html">cellProximityNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, </span>
<span>                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>,</span>
<span>                     remove_self_edges <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                     self_loop_strength <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>                     only_show_enrichment_edges <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                     rescale_edge_weights <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                     node_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                     edge_weight_range_depletion <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>                     edge_weight_range_enrichment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/27-cellProximityNetwork.png"></p>
<div class="section level3">
<h3 id="visualization-of-specific-cell-types">Visualization of specific cell types<a class="anchor" aria-label="anchor" href="#visualization-of-specific-cell-types"></a>
</h3>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Option 1</span></span>
<span><span class="va">spec_interaction</span> <span class="op">=</span> <span class="st">"cell D--cell F"</span></span>
<span></span>
<span><span class="fu"><a href="../reference/cellProximitySpatPlot2D.html">cellProximitySpatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                        interaction_name <span class="op">=</span> <span class="va">spec_interaction</span>,</span>
<span>                        show_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                        cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                        cell_color <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                        cell_color_code <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cell D'</span> <span class="op">=</span> <span class="st">'lightblue'</span>, <span class="st">'cell F'</span> <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span>,</span>
<span>                        point_size_select <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                        point_size_other <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/28-cellProximitySpatPlot2D.png"></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Option 2: create additional metadata</span></span>
<span><span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/addCellIntMetadata.html">addCellIntMetadata</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,</span>
<span>                                  spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                                  spatial_network <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>                                  cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                                  cell_interaction <span class="op">=</span> <span class="va">spec_interaction</span>,</span>
<span>                                  name <span class="op">=</span> <span class="st">'D_F_interactions'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">'D_F_interactions'</span>, </span>
<span>         legend_symbol_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>         select_cell_groups <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'other_cell D'</span>, <span class="st">'other_cell F'</span>, <span class="st">'select_cell D'</span>, <span class="st">'select_cell F'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/29-spatPlot2D.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="cell-neighborhood-interaction-changed-features">13. Cell neighborhood: interaction changed features<a class="anchor" aria-label="anchor" href="#cell-neighborhood-interaction-changed-features"></a>
</h2>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## select top 25 highest expressing genes</span></span>
<span><span class="va">gene_metadata</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/fDataDT.html">fDataDT</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gene_metadata</span><span class="op">$</span><span class="va">nr_cells</span>, <span class="va">gene_metadata</span><span class="op">$</span><span class="va">mean_expr</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gene_metadata</span><span class="op">$</span><span class="va">nr_cells</span>, <span class="va">gene_metadata</span><span class="op">$</span><span class="va">mean_expr_det</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">gene_metadata</span><span class="op">$</span><span class="va">mean_expr_det</span><span class="op">)</span></span>
<span></span>
<span><span class="va">high_expressed_genes</span> <span class="op">=</span> <span class="va">gene_metadata</span><span class="op">[</span><span class="va">mean_expr_det</span> <span class="op">&gt;</span> <span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">feat_ID</span></span>
<span></span>
<span><span class="co">## identify features (genes) that are associated with proximity to other cell types</span></span>
<span><span class="va">ICFscoresHighGenes</span> <span class="op">=</span> <span class="fu"><a href="../reference/findICF.html">findICF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                             selected_feats <span class="op">=</span> <span class="va">high_expressed_genes</span>,</span>
<span>                             spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>                             cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                             diff_test <span class="op">=</span> <span class="st">'permutation'</span>,</span>
<span>                             adjust_method <span class="op">=</span> <span class="st">'fdr'</span>,</span>
<span>                             nr_permutations <span class="op">=</span> <span class="fl">500</span>,</span>
<span>                             do_parallel <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## visualize all genes</span></span>
<span><span class="fu"><a href="../reference/plotCellProximityFeats.html">plotCellProximityFeats</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, </span>
<span>                       icfObject <span class="op">=</span> <span class="va">ICFscoresHighGenes</span>, </span>
<span>                       method <span class="op">=</span> <span class="st">'dotplot'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/30-plotCellProximityGenes.png"></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## filter genes</span></span>
<span><span class="va">ICFscoresFilt</span> <span class="op">=</span> <span class="fu"><a href="../reference/filterICF.html">filterICF</a></span><span class="op">(</span><span class="va">ICFscoresHighGenes</span>, </span>
<span>                          min_cells <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                          min_int_cells <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                          min_fdr <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                          min_spat_diff <span class="op">=</span> <span class="fl">0.1</span>, </span>
<span>                          min_log2_fc <span class="op">=</span> <span class="fl">0.1</span>, </span>
<span>                          min_zscore <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## visualize subset of interaction changed genes (ICGs)</span></span>
<span><span class="va">ICF_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Cpne2'</span>, <span class="st">'Scg3'</span>, <span class="st">'Cmtm3'</span>, <span class="st">'Cplx1'</span>, <span class="st">'Lingo1'</span><span class="op">)</span></span>
<span><span class="va">ICF_genes_types</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cell E'</span>, <span class="st">'cell D'</span>, <span class="st">'cell D'</span>, <span class="st">'cell G'</span>, <span class="st">'cell E'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ICF_genes</span><span class="op">)</span> <span class="op">=</span> <span class="va">ICF_genes_types</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotICF.html">plotICF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>        icfObject <span class="op">=</span> <span class="va">ICFscoresHighGenes</span>,</span>
<span>        source_type <span class="op">=</span> <span class="st">'cell A'</span>,</span>
<span>        source_markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Csf1r'</span>, <span class="st">'Laptm5'</span><span class="op">)</span>,</span>
<span>        ICF_feats <span class="op">=</span> <span class="va">ICF_genes</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/31-plotICF.png"></p>
</div>
<div class="section level2">
<h2 id="cell-neighborhood-ligand-receptor-cell-cell-communication">14. Cell neighborhood: ligand-receptor cell-cell communication<a class="anchor" aria-label="anchor" href="#cell-neighborhood-ligand-receptor-cell-cell-communication"></a>
</h2>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LR_data</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"Mini_datasets/seqfish/Raw/mouse_ligand_receptors.txt"</span>, </span>
<span>                                        package <span class="op">=</span> <span class="st">"GiottoData"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LR_data</span><span class="op">[</span>, <span class="va">ligand_det</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">mouseLigand</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">seqfish_mini</span><span class="op">@</span><span class="va">feat_ID</span><span class="op">[[</span><span class="st">'rna'</span><span class="op">]</span><span class="op">]</span>, <span class="cn">T</span>, <span class="cn">F</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">LR_data</span><span class="op">[</span>, <span class="va">receptor_det</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">mouseReceptor</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">seqfish_mini</span><span class="op">@</span><span class="va">feat_ID</span><span class="op">[[</span><span class="st">'rna'</span><span class="op">]</span><span class="op">]</span>, <span class="cn">T</span>, <span class="cn">F</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">LR_data_det</span> <span class="op">=</span> <span class="va">LR_data</span><span class="op">[</span><span class="va">ligand_det</span> <span class="op">==</span> <span class="cn">T</span> <span class="op">&amp;</span> <span class="va">receptor_det</span> <span class="op">==</span> <span class="cn">T</span><span class="op">]</span></span>
<span></span>
<span><span class="va">select_ligands</span> <span class="op">=</span> <span class="va">LR_data_det</span><span class="op">$</span><span class="va">mouseLigand</span></span>
<span></span>
<span><span class="va">select_receptors</span> <span class="op">=</span> <span class="va">LR_data_det</span><span class="op">$</span><span class="va">mouseReceptor</span></span>
<span></span>
<span><span class="co">## get statistical significance of gene pair expression changes based on expression ##</span></span>
<span><span class="va">expr_only_scores</span> <span class="op">=</span> <span class="fu"><a href="../reference/exprCellCellcom.html">exprCellCellcom</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                                   cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                                   random_iter <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                                   feat_set_1 <span class="op">=</span> <span class="va">select_ligands</span>,</span>
<span>                                   feat_set_2 <span class="op">=</span> <span class="va">select_receptors</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## get statistical significance of gene pair expression changes upon cell-cell interaction</span></span>
<span><span class="va">spatial_all_scores</span> <span class="op">=</span> <span class="fu"><a href="../reference/spatCellCellcom.html">spatCellCellcom</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,</span>
<span>                                     spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                                     feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>                                     spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>                                     cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                                     random_iter <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                                     feat_set_1 <span class="op">=</span> <span class="va">select_ligands</span>,</span>
<span>                                     feat_set_2 <span class="op">=</span> <span class="va">select_receptors</span>,</span>
<span>                                     adjust_method <span class="op">=</span> <span class="st">'fdr'</span>,</span>
<span>                                     do_parallel <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                     cores <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                                     verbose <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## * plot communication scores ####</span></span>
<span><span class="co">## select top LR ##</span></span>
<span><span class="va">selected_spat</span> <span class="op">=</span> <span class="va">spatial_all_scores</span><span class="op">[</span><span class="va">p.adj</span> <span class="op">&lt;=</span> <span class="fl">0.5</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">log2fc</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.1</span> <span class="op">&amp;</span> <span class="va">lig_nr</span> <span class="op">&gt;=</span> <span class="fl">2</span> <span class="op">&amp;</span> <span class="va">rec_nr</span> <span class="op">&gt;=</span> <span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">setorder</a></span><span class="op">(</span><span class="va">selected_spat</span>, <span class="op">-</span><span class="va">PI</span><span class="op">)</span></span>
<span></span>
<span><span class="va">top_LR_ints</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">selected_spat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">PI</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">$</span><span class="va">LR_comb</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">33</span><span class="op">]</span></span>
<span></span>
<span><span class="va">top_LR_cell_ints</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">selected_spat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">PI</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">$</span><span class="va">LR_cell_comb</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">33</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotCCcomHeatmap.html">plotCCcomHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                 comScores <span class="op">=</span> <span class="va">spatial_all_scores</span>,</span>
<span>                 selected_LR <span class="op">=</span> <span class="va">top_LR_ints</span>,</span>
<span>                 selected_cell_LR <span class="op">=</span> <span class="va">top_LR_cell_ints</span>,</span>
<span>                 show <span class="op">=</span> <span class="st">'LR_expr'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/32-plotCCcomHeatmap.png"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCCcomDotplot.html">plotCCcomDotplot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                 comScores <span class="op">=</span> <span class="va">spatial_all_scores</span>,</span>
<span>                 selected_LR <span class="op">=</span> <span class="va">top_LR_ints</span>,</span>
<span>                 selected_cell_LR <span class="op">=</span> <span class="va">top_LR_cell_ints</span>,</span>
<span>                 cluster_on <span class="op">=</span> <span class="st">'PI'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/33-plotCCcomDotplot.png"></p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## * spatial vs rank ####</span></span>
<span><span class="va">comb_comm</span> <span class="op">=</span> <span class="fu"><a href="../reference/combCCcom.html">combCCcom</a></span><span class="op">(</span>spatialCC <span class="op">=</span> <span class="va">spatial_all_scores</span>,</span>
<span>                      exprCC <span class="op">=</span> <span class="va">expr_only_scores</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># top differential activity levels for ligand receptor pairs</span></span>
<span><span class="fu"><a href="../reference/plotRankSpatvsExpr.html">plotRankSpatvsExpr</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>                   <span class="va">comb_comm</span>,</span>
<span>                   expr_rnk_column <span class="op">=</span> <span class="st">'exprPI_rnk'</span>,</span>
<span>                   spat_rnk_column <span class="op">=</span> <span class="st">'spatPI_rnk'</span>,</span>
<span>                   gradient_midpoint <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/34-plotRankSpatvsExpr.png"></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## * recovery ####</span></span>
<span><span class="co">## predict maximum differential activity</span></span>
<span><span class="fu"><a href="../reference/plotRecovery.html">plotRecovery</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,</span>
<span>             <span class="va">comb_comm</span>,</span>
<span>             expr_rnk_column <span class="op">=</span> <span class="st">'exprPI_rnk'</span>,</span>
<span>             spat_rnk_column <span class="op">=</span> <span class="st">'spatPI_rnk'</span>,</span>
<span>             ground_truth <span class="op">=</span> <span class="st">'spatial'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mini_seqfish/35-plotRecovery.png"></p>
</div>
<div class="section level2">
<h2 id="session-info">15. Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">3.2</span> (<span class="dv">2023-10-31</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>apple<span class="sc">-</span><span class="fu">darwin20</span> (<span class="dv">64</span><span class="sc">-</span>bit)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="fl">14.3</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.3</span><span class="sc">-</span>x86_64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] smfishHmrf_0<span class="fl">.1</span>     fs_1.<span class="fl">6.3</span>           pracma_2.<span class="fl">4.4</span>       ggdendro_0.<span class="fl">1.23</span>   </span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a>[<span class="dv">5</span>] GiottoData_0.<span class="dv">2</span>.<span class="fl">6.2</span> GiottoUtils_0.<span class="fl">1.3</span>  Giotto_4.<span class="fl">0.2</span>       GiottoClass_0.<span class="fl">1.3</span> </span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">1</span>] RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>          rstudioapi_0.<span class="fl">15.0</span>           jsonlite_1.<span class="fl">8.8</span>             </span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">4</span>] shape_1.<span class="fl">4.6</span>                 magrittr_2.<span class="fl">0.3</span>              magick_2.<span class="fl">8.2</span>               </span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">7</span>] farver_2.<span class="fl">1.1</span>                GlobalOptions_0.<span class="fl">1.2</span>         zlibbioc_1.<span class="fl">48.0</span>            </span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a> [<span class="dv">10</span>] ragg_1.<span class="fl">2.7</span>                  vctrs_0.<span class="fl">6.5</span>                 Cairo_1<span class="fl">.6</span><span class="dv">-2</span>                </span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a> [<span class="dv">13</span>] RCurl_1<span class="fl">.98-1.14</span>             terra_1<span class="fl">.7</span><span class="dv">-71</span>                S4Arrays_1.<span class="fl">2.0</span>             </span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a> [<span class="dv">16</span>] SparseArray_1.<span class="fl">2.3</span>           parallelly_1.<span class="fl">36.0</span>           plyr_1.<span class="fl">8.9</span>                 </span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a> [<span class="dv">19</span>] igraph_2.<span class="dv">0</span>.<span class="fl">1.1</span>              lifecycle_1.<span class="fl">0.4</span>             iterators_1.<span class="fl">0.14</span>           </span>
<span id="cb43-30"><a href="#cb43-30" aria-hidden="true" tabindex="-1"></a> [<span class="dv">22</span>] pkgconfig_2.<span class="fl">0.3</span>             rsvd_1.<span class="fl">0.5</span>                  Matrix_1<span class="fl">.6</span><span class="dv">-5</span>               </span>
<span id="cb43-31"><a href="#cb43-31" aria-hidden="true" tabindex="-1"></a> [<span class="dv">25</span>] R6_2.<span class="fl">5.1</span>                    GenomeInfoDbData_1.<span class="fl">2.11</span>     rbibutils_2.<span class="fl">2.16</span>           </span>
<span id="cb43-32"><a href="#cb43-32" aria-hidden="true" tabindex="-1"></a> [<span class="dv">28</span>] MatrixGenerics_1.<span class="fl">14.0</span>       future_1.<span class="fl">33.1</span>               clue_0<span class="fl">.3</span><span class="dv">-65</span>                </span>
<span id="cb43-33"><a href="#cb43-33" aria-hidden="true" tabindex="-1"></a> [<span class="dv">31</span>] digest_0.<span class="fl">6.34</span>               colorspace_2<span class="fl">.1</span><span class="dv">-0</span>            S4Vectors_0.<span class="fl">40.2</span>           </span>
<span id="cb43-34"><a href="#cb43-34" aria-hidden="true" tabindex="-1"></a> [<span class="dv">34</span>] irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>               textshaping_0.<span class="fl">3.7</span>           GenomicRanges_1.<span class="fl">54.1</span>       </span>
<span id="cb43-35"><a href="#cb43-35" aria-hidden="true" tabindex="-1"></a> [<span class="dv">37</span>] beachmat_2.<span class="fl">18.0</span>             labeling_0.<span class="fl">4.3</span>              progressr_0.<span class="fl">14.0</span>           </span>
<span id="cb43-36"><a href="#cb43-36" aria-hidden="true" tabindex="-1"></a> [<span class="dv">40</span>] fansi_1.<span class="fl">0.6</span>                 polyclip_1<span class="fl">.10</span><span class="dv">-6</span>             abind_1<span class="fl">.4</span><span class="dv">-5</span>                </span>
<span id="cb43-37"><a href="#cb43-37" aria-hidden="true" tabindex="-1"></a> [<span class="dv">43</span>] compiler_4.<span class="fl">3.2</span>              withr_3.<span class="fl">0.0</span>                 doParallel_1.<span class="fl">0.17</span>          </span>
<span id="cb43-38"><a href="#cb43-38" aria-hidden="true" tabindex="-1"></a> [<span class="dv">46</span>] backports_1.<span class="fl">4.1</span>             BiocParallel_1.<span class="fl">36.0</span>         viridis_0.<span class="fl">6.5</span>              </span>
<span id="cb43-39"><a href="#cb43-39" aria-hidden="true" tabindex="-1"></a> [<span class="dv">49</span>] ggforce_0.<span class="fl">4.1</span>               MASS_7<span class="fl">.3</span><span class="dv">-60</span>.<span class="fl">0.1</span>             DelayedArray_0.<span class="fl">28.0</span>        </span>
<span id="cb43-40"><a href="#cb43-40" aria-hidden="true" tabindex="-1"></a> [<span class="dv">52</span>] rjson_0.<span class="fl">2.21</span>                gtools_3.<span class="fl">9.5</span>                GiottoVisuals_0.<span class="fl">1.4</span>        </span>
<span id="cb43-41"><a href="#cb43-41" aria-hidden="true" tabindex="-1"></a> [<span class="dv">55</span>] tools_4.<span class="fl">3.2</span>                 future.apply_1.<span class="fl">11.1</span>         glue_1.<span class="fl">7.0</span>                 </span>
<span id="cb43-42"><a href="#cb43-42" aria-hidden="true" tabindex="-1"></a> [<span class="dv">58</span>] dbscan_1<span class="fl">.1</span><span class="dv">-12</span>               grid_4.<span class="fl">3.2</span>                  checkmate_2.<span class="fl">3.1</span>            </span>
<span id="cb43-43"><a href="#cb43-43" aria-hidden="true" tabindex="-1"></a> [<span class="dv">61</span>] Rtsne_0<span class="fl">.17</span>                  cluster_2.<span class="fl">1.6</span>               reshape2_1.<span class="fl">4.4</span>             </span>
<span id="cb43-44"><a href="#cb43-44" aria-hidden="true" tabindex="-1"></a> [<span class="dv">64</span>] generics_0.<span class="fl">1.3</span>              gtable_0.<span class="fl">3.4</span>                tidyr_1.<span class="fl">3.1</span>                </span>
<span id="cb43-45"><a href="#cb43-45" aria-hidden="true" tabindex="-1"></a> [<span class="dv">67</span>] data.table_1.<span class="fl">15.0</span>           BiocSingular_1.<span class="fl">18.0</span>         tidygraph_1.<span class="fl">3.1</span>            </span>
<span id="cb43-46"><a href="#cb43-46" aria-hidden="true" tabindex="-1"></a> [<span class="dv">70</span>] ScaledMatrix_1.<span class="fl">10.0</span>         utf8_1.<span class="fl">2.4</span>                  XVector_0.<span class="fl">42.0</span>             </span>
<span id="cb43-47"><a href="#cb43-47" aria-hidden="true" tabindex="-1"></a> [<span class="dv">73</span>] BiocGenerics_0.<span class="fl">48.1</span>         ggrepel_0.<span class="fl">9.5</span>               foreach_1.<span class="fl">5.2</span>              </span>
<span id="cb43-48"><a href="#cb43-48" aria-hidden="true" tabindex="-1"></a> [<span class="dv">76</span>] pillar_1.<span class="fl">9.0</span>                stringr_1.<span class="fl">5.1</span>               limma_3.<span class="fl">58.1</span>               </span>
<span id="cb43-49"><a href="#cb43-49" aria-hidden="true" tabindex="-1"></a> [<span class="dv">79</span>] circlize_0.<span class="fl">4.15</span>             dplyr_1.<span class="fl">1.4</span>                 tweenr_2.<span class="fl">0.2</span>               </span>
<span id="cb43-50"><a href="#cb43-50" aria-hidden="true" tabindex="-1"></a> [<span class="dv">82</span>] lattice_0<span class="fl">.22</span><span class="dv">-5</span>              FNN_1.<span class="fl">1.4</span>                   deldir_2<span class="fl">.0</span><span class="dv">-2</span>               </span>
<span id="cb43-51"><a href="#cb43-51" aria-hidden="true" tabindex="-1"></a> [<span class="dv">85</span>] tidyselect_1.<span class="fl">2.0</span>            ComplexHeatmap_2.<span class="fl">18.0</span>       SingleCellExperiment_1.<span class="fl">24.0</span></span>
<span id="cb43-52"><a href="#cb43-52" aria-hidden="true" tabindex="-1"></a> [<span class="dv">88</span>] knitr_1<span class="fl">.45</span>                  gridExtra_2<span class="fl">.3</span>               IRanges_2.<span class="fl">36.0</span>             </span>
<span id="cb43-53"><a href="#cb43-53" aria-hidden="true" tabindex="-1"></a> [<span class="dv">91</span>] SummarizedExperiment_1.<span class="fl">32.0</span> stats4_4.<span class="fl">3.2</span>                xfun_0<span class="fl">.41</span>                  </span>
<span id="cb43-54"><a href="#cb43-54" aria-hidden="true" tabindex="-1"></a> [<span class="dv">94</span>] graphlayouts_1.<span class="fl">1.0</span>          Biobase_2.<span class="fl">62.0</span>              statmod_1.<span class="fl">5.0</span>              </span>
<span id="cb43-55"><a href="#cb43-55" aria-hidden="true" tabindex="-1"></a> [<span class="dv">97</span>] matrixStats_1.<span class="fl">2.0</span>           stringi_1.<span class="fl">8.3</span>               codetools_0<span class="fl">.2</span><span class="dv">-19</span>           </span>
<span id="cb43-56"><a href="#cb43-56" aria-hidden="true" tabindex="-1"></a>[<span class="dv">100</span>] ggraph_2.<span class="fl">1.0</span>                tibble_3.<span class="fl">2.1</span>                colorRamp2_0.<span class="fl">1.0</span>           </span>
<span id="cb43-57"><a href="#cb43-57" aria-hidden="true" tabindex="-1"></a>[<span class="dv">103</span>] cli_3.<span class="fl">6.2</span>                   uwot_0.<span class="fl">1.16</span>                 reticulate_1.<span class="fl">35.0</span>          </span>
<span id="cb43-58"><a href="#cb43-58" aria-hidden="true" tabindex="-1"></a>[<span class="dv">106</span>] systemfonts_1.<span class="fl">0.5</span>           Rdpack_2<span class="fl">.6</span>                  munsell_0.<span class="fl">5.0</span>              </span>
<span id="cb43-59"><a href="#cb43-59" aria-hidden="true" tabindex="-1"></a>[<span class="dv">109</span>] Rcpp_1.<span class="fl">0.12</span>                 GenomeInfoDb_1.<span class="fl">38.5</span>         globals_0.<span class="fl">16.2</span>             </span>
<span id="cb43-60"><a href="#cb43-60" aria-hidden="true" tabindex="-1"></a>[<span class="dv">112</span>] png_0<span class="fl">.1</span><span class="dv">-8</span>                   parallel_4.<span class="fl">3.2</span>              ggplot2_3.<span class="fl">4.4</span>              </span>
<span id="cb43-61"><a href="#cb43-61" aria-hidden="true" tabindex="-1"></a>[<span class="dv">115</span>] bitops_1<span class="fl">.0</span><span class="dv">-7</span>                listenv_0.<span class="fl">9.1</span>               SpatialExperiment_1.<span class="fl">12.0</span>   </span>
<span id="cb43-62"><a href="#cb43-62" aria-hidden="true" tabindex="-1"></a>[<span class="dv">118</span>] viridisLite_0.<span class="fl">4.2</span>           scales_1.<span class="fl">3.0</span>                purrr_1.<span class="fl">0.2</span>                </span>
<span id="cb43-63"><a href="#cb43-63" aria-hidden="true" tabindex="-1"></a>[<span class="dv">121</span>] crayon_1.<span class="fl">5.2</span>                GetoptLong_1.<span class="fl">0.5</span>            rlang_1.<span class="fl">1.3</span>                </span>
<span id="cb43-64"><a href="#cb43-64" aria-hidden="true" tabindex="-1"></a>[<span class="dv">124</span>] cowplot_1.<span class="fl">1.3</span> </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. ChÃ¡vez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew Oâ€™Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
