<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Nanostring CosMx Subcellular Lung Cancer • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Nanostring CosMx Subcellular Lung Cancer">
<meta property="og:description" content="Giotto">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
    <li>
      <a href="../articles/contributing.html">Contributing guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html">Spatial geometries</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Nanostring CosMx Subcellular Lung Cancer</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/HEAD/vignettes/nanostring_cosmx_lung_cancer.Rmd" class="external-link"><code>vignettes/nanostring_cosmx_lung_cancer.Rmd</code></a></small>
      <div class="hidden name"><code>nanostring_cosmx_lung_cancer.Rmd</code></div>

    </div>

    
    
<p>This example uses subcellular data from Nanostring’s CosMx Spatial
Molecular Imager. This publicly available <a href="https://nanostring.com/resources/smi-ffpe-dataset-lung12-data/" class="external-link">dataset</a>
is from an FFPE sample of non-small-cell lung cancer (NSCLC). This
example works with Lung12.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed.</span></span>
<span><span class="va">genv_exists</span> <span class="op">=</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/checkGiottoEnvironment.html">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/installGiottoEnvironment.html">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="section level2">
<h2 id="setup">1. Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Custom color palettes from rcartocolor</span></span>
<span><span class="co"># pal10 = rcartocolor::carto_pal(n = 10, name = 'Pastel')</span></span>
<span><span class="va">pal10</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#66C5CC"</span>,<span class="st">"#F6CF71"</span>,<span class="st">"#F89C74"</span>,<span class="st">"#DCB0F2"</span>,<span class="st">"#87C55F"</span>,</span>
<span>          <span class="st">"#9EB9F3"</span>,<span class="st">"#FE88B1"</span>,<span class="st">"#C9DB74"</span>,<span class="st">"#8BE0A4"</span>,<span class="st">"#B3B3B3"</span><span class="op">)</span></span>
<span><span class="co"># viv10 = rcartocolor::carto_pal(n = 10, name = 'Vivid')</span></span>
<span><span class="va">viv10</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#E58606"</span>,<span class="st">"#5D69B1"</span>,<span class="st">"#52BCA3"</span>,<span class="st">"#99C945"</span>,<span class="st">"#CC61B0"</span>,</span>
<span>          <span class="st">"#24796C"</span>,<span class="st">"#DAA51B"</span>,<span class="st">"#2F8AC4"</span>,<span class="st">"#764E9F"</span>,<span class="st">"#A5AA99"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set working directory</span></span>
<span><span class="va">results_folder</span> <span class="op">=</span> <span class="st">'/path/to/directory'</span></span>
<span></span>
<span><span class="co"># Optional: Specify a path to a Python executable within a conda or miniconda</span></span>
<span><span class="co"># environment. If set to NULL (default), the Python executable within the previously</span></span>
<span><span class="co"># installed Giotto environment will be used.</span></span>
<span><span class="va">my_python_path</span> <span class="op">=</span> <span class="cn">NULL</span> <span class="co"># alternatively, "/local/python/path/python" if desired.</span></span>
<span></span>
<span><span class="co">## Set object behavior</span></span>
<span><span class="co"># by directly saving plots, but not rendering them you will save a lot of time</span></span>
<span><span class="va">instrs</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoInstructions.html">createGiottoInstructions</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">results_folder</span>,</span>
<span>                                  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  show_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                  return_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                  python_path <span class="op">=</span> <span class="va">my_python_path</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="cosmx-project-loading-function">1.1 CosMx Project loading function<a class="anchor" aria-label="anchor" href="#cosmx-project-loading-function"></a>
</h3>
<p>Convenience function for loading in the CosMx data. It loads
subcellular transcript information and polygons and generates a
<code>giotto</code> object with <code>giottoPoints</code> objects for
both ‘rna’ and ‘neg_probe’ nested in the gobject <code>feat_info</code>
slot, and a <code>giottoPolygon</code> object for the ‘cell’ spatial
unit in the <code>spatial_info</code> slot.</p>
<p>This function performs the manual object creation steps described
below. To skip those steps and preliminary data exploration, go to
Section 5.</p>
<p>Additionally, a comparison of the count matrix produced through the
convenience function ‘subcellular’ workflow and Nanostring’s provided
matrix can be found at Section 6.4.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## provide path to nanostring folder</span></span>
<span><span class="va">data_path</span> <span class="op">=</span> <span class="st">'/path/to/data/Lung12-Flat_files_and_images/'</span></span>
<span></span>
<span><span class="co">## create giotto cosmx object</span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="../reference/createGiottoCosMxObject.html">createGiottoCosMxObject</a></span><span class="op">(</span>cosmx_dir <span class="op">=</span> <span class="va">data_path</span>,</span>
<span>                                   data_to_use <span class="op">=</span> <span class="st">'subcellular'</span>, <span class="co"># only subcellular</span></span>
<span>                                   FOVs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span>,</span>
<span>                                   instructions <span class="op">=</span> <span class="va">instrs</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoFeatInfo.html">showGiottoFeatInfo</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoSpatialInfo.html">showGiottoSpatialInfo</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="data-exploration-and-loading">2. Data exploration and loading<a class="anchor" aria-label="anchor" href="#data-exploration-and-loading"></a>
</h2>
<div class="section level3">
<h3 id="subcellular-detections-points-info">2.1 Subcellular detections (points info)<a class="anchor" aria-label="anchor" href="#subcellular-detections-points-info"></a>
</h3>
<p><code>tx_file.csv</code> contains the subcellular detections
information. It contains information on each of the individual feature
detections within the sample.</p>
<ul>
<li>
<strong>fov</strong> which FOV the detection happened in</li>
<li>
<strong>cell_ID</strong> the ID of the cell the detection happened
in</li>
<li>
<strong>x_global_px</strong> the global spatial x location in
pixels</li>
<li>
<strong>y_global_px</strong> the global spatial y location in
pixels</li>
<li>
<strong>x_local_px</strong> the spatial x location in pixels within
the FOV</li>
<li>
<strong>y_local_px</strong> the spatial y location in pixels within
the FOV</li>
<li>
<strong>z</strong> the z plane the detection was called in (-1 to
16)</li>
<li>
<strong>target</strong> the feature the probe is targeted
against</li>
<li>
<strong>CellComp</strong> Cellular compartment the detection
happened in (0, Cytoplasm, Membrane, Nuclear)</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load transcript coordinates</span></span>
<span><span class="va">tx_coord_all</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">'Lung12_tx_file.csv'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">colnames</a></span><span class="op">(</span><span class="va">tx_coord_all</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># z planes</span></span>
<span><span class="va">tx_coord_all</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Cell compartment</span></span>
<span><span class="va">tx_coord_all</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">CellComp</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># [1] "fov"         "cell_ID"     "x_global_px" "y_global_px"</span></span>
<span><span class="co"># [5] "x_local_px"  "y_local_px"  "z"           "target"     </span></span>
<span><span class="co"># [9] "CellComp" </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># z</span></span>
<span><span class="co">#      -1       0       1       2       3       4       5       6 </span></span>
<span><span class="co">#   23723 3466178 2522315 2694973 2686531 2648926 2660346 2711105 </span></span>
<span><span class="co">#       7       8       9      10      11      12      13      14 </span></span>
<span><span class="co"># 2855259 3700831   36840    6594    6466    6787    6944    6959 </span></span>
<span><span class="co">#      15      16 </span></span>
<span><span class="co">#   17603       2 </span></span>
<span><span class="co">#   </span></span>
<span><span class="co"># CellComp</span></span>
<span><span class="co">#         0 Cytoplasm  Membrane   Nuclear </span></span>
<span><span class="co">#   6619744   5770549   3368411  10299678</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="split-detections-by-features-vs-negative-probes">2.2 Split detections by features vs negative probes<a class="anchor" aria-label="anchor" href="#split-detections-by-features-vs-negative-probes"></a>
</h3>
<p><code>tx_file.csv</code> contains information on both actual features
(960 targeted gene probes in this dataset) and negative probes (20) that
are targeted to alien sequences defined by the External RNA Controls
Consortium (ERCC) that do not exist in human tissue. These two types of
detections will be treated as separate feature types
(<code>feat_type</code>) and placed in separate expression matrices.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_IDs</span> <span class="op">=</span> <span class="va">tx_coord_all</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">target</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># negative probe IDs</span></span>
<span><span class="va">neg_IDs</span> <span class="op">=</span> <span class="va">all_IDs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">'NegPrb'</span>, <span class="va">all_IDs</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">neg_IDs</span></span>
<span></span>
<span><span class="co"># Feature IDs</span></span>
<span><span class="va">feat_IDs</span> <span class="op">=</span> <span class="va">all_IDs</span><span class="op">[</span><span class="op">!</span><span class="va">all_IDs</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">neg_IDs</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">feat_IDs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># split detections</span></span>
<span><span class="va">feat_coords_all</span> <span class="op">=</span> <span class="va">tx_coord_all</span><span class="op">[</span><span class="va">target</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">feat_IDs</span><span class="op">]</span></span>
<span><span class="va">neg_coords_all</span> <span class="op">=</span> <span class="va">tx_coord_all</span><span class="op">[</span><span class="va">target</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">neg_IDs</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">'\nFeatures: '</span>, <span class="va">feat_coords_all</span><span class="op">[</span>, <span class="va">.N</span><span class="op">]</span>, <span class="st">'\n'</span>,</span>
<span>    <span class="st">'NegProbes: '</span>, <span class="va">neg_coords_all</span><span class="op">[</span>, <span class="va">.N</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Negative Probe IDs</span></span>
<span><span class="co">#  [1] "NegPrb15" "NegPrb18" "NegPrb7"  "NegPrb21" "NegPrb13"</span></span>
<span><span class="co">#  [6] "NegPrb10" "NegPrb11" "NegPrb9"  "NegPrb3"  "NegPrb16"</span></span>
<span><span class="co"># [11] "NegPrb23" "NegPrb14" "NegPrb20" "NegPrb8"  "NegPrb19"</span></span>
<span><span class="co"># [16] "NegPrb6"  "NegPrb5"  "NegPrb12" "NegPrb17" "NegPrb22"</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># Number of feature IDs</span></span>
<span><span class="co"># [1] 960</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># Features:  25875734</span></span>
<span><span class="co"># NegProbes:  182648</span></span></code></pre></div>
<div class="section level4">
<h4 id="preview-negative-probes-optional">2.2.1 Preview negative probes (optional)<a class="anchor" aria-label="anchor" href="#preview-negative-probes-optional"></a>
</h4>
<p>Previewing the probe information can be done by converting to
<code>giottoPoints</code> and then using <code><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot()</a></code>. Here we
show a preview of the negative probes.</p>
<p><strong>Note:</strong> if previewing the rna expression information,
it is highly recommended to set a subset of features using the
<code>feats</code> param. The default is to plot all points, which can
be very slow for large data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">neg_points</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPoints.html">createGiottoPoints</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">neg_coords_all</span><span class="op">[</span>, <span class="fu">.</span><span class="op">(</span><span class="va">target</span>, <span class="va">x_global_px</span>, <span class="va">y_global_px</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">neg_points</span>, point_size <span class="op">=</span> <span class="fl">0.2</span>, feats <span class="op">=</span> <span class="va">neg_IDs</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/preview_negprb.png"></p>
</div>
</div>
<div class="section level3">
<h3 id="fov-shifts">2.3 FOV shifts<a class="anchor" aria-label="anchor" href="#fov-shifts"></a>
</h3>
<p><code>fov_positions_file.csv</code> contains information on the x and
y shifts needed in order to put the FOVs tiles together into a cohesive
whole. This information is needed during the image attachment and
alignment process.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#  load field of vision (fov) positions</span></span>
<span><span class="va">fov_offset_file</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, </span>
<span>                                           <span class="st">'Lung12_fov_positions_file.csv'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="choose-field-of-view-for-analysis">2.4 Choose field of view for analysis<a class="anchor" aria-label="anchor" href="#choose-field-of-view-for-analysis"></a>
</h3>
<p>CosMx data is large and Giotto loads in the subcellular information
by FOV. This dataset includes 28 FOVs which can be difficult for most
computers to handle at once.</p>
<p>This tutorial will use FOVs ‘02’, ‘03’, and ‘04’ which correspond to
the 3 FOVs visible on the bottom right in the negative probe preview
above.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gobjects_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">id_set</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'02'</span>, <span class="st">'03'</span>, <span class="st">'04'</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="create-a-giotto-object-for-each-fov">3. Create a Giotto Object for each FOV<a class="anchor" aria-label="anchor" href="#create-a-giotto-object-for-each-fov"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span><span class="op">(</span><span class="va">fov_i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">id_set</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>  <span class="va">fov_id</span> <span class="op">=</span> <span class="va">id_set</span><span class="op">[</span><span class="va">fov_i</span><span class="op">]</span></span>
<span></span>
<span></span>
<span>  <span class="co"># 1. original composite image as png</span></span>
<span>  <span class="va">original_composite_image</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">'CellComposite/CellComposite_F0'</span>, <span class="va">fov_id</span>,<span class="st">'.jpg'</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># 2. input cell segmentation as mask file</span></span>
<span>  <span class="va">segmentation_mask</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">'CellLabels/CellLabels_F0'</span>, <span class="va">fov_id</span>, <span class="st">'.tif'</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># 3. input features coordinates + offset</span></span>
<span>  <span class="va">feat_coord</span> <span class="op">=</span> <span class="va">feat_coords_all</span><span class="op">[</span><span class="va">fov</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">fov_id</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="va">neg_coord</span> <span class="op">=</span> <span class="va">neg_coords_all</span><span class="op">[</span><span class="va">fov</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">fov_id</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="va">feat_coord</span> <span class="op">=</span> <span class="va">feat_coord</span><span class="op">[</span>,<span class="fu">.</span><span class="op">(</span><span class="va">x_local_px</span>, <span class="va">y_local_px</span>, <span class="va">z</span>, <span class="va">target</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="va">neg_coord</span> <span class="op">=</span> <span class="va">neg_coord</span><span class="op">[</span>,<span class="fu">.</span><span class="op">(</span><span class="va">x_local_px</span>, <span class="va">y_local_px</span>, <span class="va">z</span>, <span class="va">target</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">colnames</a></span><span class="op">(</span><span class="va">feat_coord</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'x'</span>, <span class="st">'y'</span>, <span class="st">'z'</span>, <span class="st">'gene_id'</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">colnames</a></span><span class="op">(</span><span class="va">neg_coord</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'x'</span>, <span class="st">'y'</span>, <span class="st">'z'</span>, <span class="st">'gene_id'</span><span class="op">)</span></span>
<span>  <span class="va">feat_coord</span> <span class="op">=</span> <span class="va">feat_coord</span><span class="op">[</span>,<span class="fu">.</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">gene_id</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="va">neg_coord</span> <span class="op">=</span> <span class="va">neg_coord</span><span class="op">[</span>,<span class="fu">.</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">gene_id</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span></span>
<span>  <span class="va">fovsubset</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObjectSubcellular.html">createGiottoObjectSubcellular</a></span><span class="op">(</span></span>
<span>    gpoints <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'rna'</span> <span class="op">=</span> <span class="va">feat_coord</span>,</span>
<span>                   <span class="st">'neg_probe'</span> <span class="op">=</span> <span class="va">neg_coord</span><span class="op">)</span>,</span>
<span>    gpolygons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'cell'</span> <span class="op">=</span> <span class="va">segmentation_mask</span><span class="op">)</span>,</span>
<span>    polygon_mask_list_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>      mask_method <span class="op">=</span> <span class="st">'guess'</span>,</span>
<span>      flip_vertical <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      flip_horizontal <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>      shift_horizontal_step <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>    <span class="op">)</span>,</span>
<span>    instructions <span class="op">=</span> <span class="va">instrs</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span></span>
<span>  <span class="co"># cell centroids are now used to provide the spatial locations</span></span>
<span>  <span class="va">fovsubset</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addSpatialCentroidLocations.html">addSpatialCentroidLocations</a></span><span class="op">(</span><span class="va">fovsubset</span>,</span>
<span>                                          poly_info <span class="op">=</span> <span class="st">'cell'</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># create and add Giotto images</span></span>
<span>  <span class="va">composite</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoLargeImage.html">createGiottoLargeImage</a></span><span class="op">(</span>raster_object <span class="op">=</span> <span class="va">original_composite_image</span>,</span>
<span>                                     negative_y <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                     name <span class="op">=</span> <span class="st">'composite'</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">fovsubset</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addGiottoImage.html">addGiottoImage</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fovsubset</span>,</span>
<span>                             largeImages <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">composite</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span>  <span class="va">fovsubset</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/convertGiottoLargeImageToMG.html">convertGiottoLargeImageToMG</a></span><span class="op">(</span>giottoLargeImage <span class="op">=</span> <span class="va">composite</span>,</span>
<span>                                          <span class="co">#mg_name = 'composite',</span></span>
<span>                                          gobject <span class="op">=</span> <span class="va">fovsubset</span>,</span>
<span>                                          return_gobject <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">gobjects_list</span><span class="op">[[</span><span class="va">fov_i</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="va">fovsubset</span></span>
<span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="join-fov-giotto-objects">4. Join FOV Giotto Objects<a class="anchor" aria-label="anchor" href="#join-fov-giotto-objects"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">new_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"fov0"</span>, <span class="va">id_set</span><span class="op">)</span></span>
<span></span>
<span><span class="va">id_match</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">id_set</span><span class="op">)</span>, <span class="va">fov_offset_file</span><span class="op">$</span><span class="va">fov</span><span class="op">)</span></span>
<span><span class="va">x_shifts</span> <span class="op">=</span> <span class="va">fov_offset_file</span><span class="op">[</span><span class="va">id_match</span><span class="op">]</span><span class="op">$</span><span class="va">x_global_px</span></span>
<span><span class="va">y_shifts</span> <span class="op">=</span> <span class="va">fov_offset_file</span><span class="op">[</span><span class="va">id_match</span><span class="op">]</span><span class="op">$</span><span class="va">y_global_px</span></span>
<span></span>
<span><span class="co"># Create Giotto object that includes all selected FOVs</span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html">joinGiottoObjects</a></span><span class="op">(</span>gobject_list <span class="op">=</span> <span class="va">gobjects_list</span>,</span>
<span>                             gobject_names <span class="op">=</span> <span class="va">new_names</span>,</span>
<span>                             join_method <span class="op">=</span> <span class="st">'shift'</span>,</span>
<span>                             x_shift <span class="op">=</span> <span class="va">x_shifts</span>,</span>
<span>                             y_shift <span class="op">=</span> <span class="va">y_shifts</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-cells-and-genes-of-interest">5. Visualize Cells and Genes of Interest<a class="anchor" aria-label="anchor" href="#visualize-cells-and-genes-of-interest"></a>
</h2>
<p>When plotting subcellular data, Giotto uses the
<code>spatInSituPlot</code> functions. Spatial plots showing the feature
points and polygons are plotted using
<code><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints()</a></code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoImageNames.html">showGiottoImageNames</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set up vector of image names</span></span>
<span><span class="va">id_set</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'02'</span>, <span class="st">'03'</span>, <span class="st">'04'</span><span class="op">)</span></span>
<span><span class="va">new_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"fov0"</span>, <span class="va">id_set</span><span class="op">)</span></span>
<span><span class="va">image_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">new_names</span>, <span class="st">'-image'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                     show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     image_name <span class="op">=</span> <span class="va">image_names</span>,</span>
<span>                     feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'rna'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'MMP2'</span>, <span class="st">'VEGFA'</span>, <span class="st">'IGF1R'</span>,</span>
<span>                                            <span class="st">'MKI67'</span>, <span class="st">'EPCAM'</span>, <span class="st">'KRT8'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                     feats_color_code <span class="op">=</span> <span class="va">viv10</span>,</span>
<span>                     spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                     point_size <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     use_overlap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     polygon_feat_type <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">'white'</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.03</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                                       save_name <span class="op">=</span> <span class="st">'1_inSituFeats'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/1_inSituFeats.png"></p>
<div class="section level3">
<h3 id="visualize-cell-centroids">5.1 Visualize Cell Centroids<a class="anchor" aria-label="anchor" href="#visualize-cell-centroids"></a>
</h3>
<p>The standard <code><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot2D()</a></code> function can also be used, but
this works off only the aggregated information that is assembled based
on the subcellular information. Plotting information based on cell
centroids can be done through this function.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>           image_name <span class="op">=</span> <span class="va">image_names</span>,</span>
<span>           show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           point_shape <span class="op">=</span> <span class="st">'no_border'</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>           coord_fix_ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>           save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                             save_name <span class="op">=</span> <span class="st">'2_spatCentroids'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/2_spatCentroids.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="aggregate-subcellular-features">6. Aggregate subcellular features<a class="anchor" aria-label="anchor" href="#aggregate-subcellular-features"></a>
</h2>
<p>Giotto supports working directly with the subcellular features in
order to generate cell by feature matrices. The data generated this way
is then given the spatial unit <code>'cell'</code>. This workflow is
recommended over loading the provided cell by feature (aggregated
expression) matrix and then including the subcellular information as
secondary data.</p>
<p>When both the raw subcellular information and the pre-made expression
matrix are loaded in at the same time, the subcellular data and all data
generated from it should be given the spatial unit <code>'cell'</code>
and the pre-generated aggregated information should be given a different
spatial unit such as <code>'cell_agg'</code> to differentiate between
the two sources of information.</p>
<p>In this step, we will be aggregating the feature points of
<code>'rna'</code> and <code>'neg_probe'</code> into the
<code>'cell'</code> spatial unit.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find the feature points overlapped by polygons. This overlap information is then</span></span>
<span><span class="co"># returned to the relevant giottoPolygon object's overlaps slot.</span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/calculateOverlapRaster.html">calculateOverlapRaster</a></span><span class="op">(</span><span class="va">fov_join</span>, feat_info <span class="op">=</span> <span class="st">'rna'</span><span class="op">)</span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/calculateOverlapRaster.html">calculateOverlapRaster</a></span><span class="op">(</span><span class="va">fov_join</span>, feat_info <span class="op">=</span> <span class="st">'neg_probe'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert the overlap information into a cell by feature expression matrix which</span></span>
<span><span class="co"># is then stored in the Giotto object's expression slot</span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/overlapToMatrix.html">overlapToMatrix</a></span><span class="op">(</span><span class="va">fov_join</span>, feat_info <span class="op">=</span> <span class="st">'rna'</span><span class="op">)</span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/overlapToMatrix.html">overlapToMatrix</a></span><span class="op">(</span><span class="va">fov_join</span>, feat_info <span class="op">=</span> <span class="st">'neg_probe'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoExpression.html">showGiottoExpression</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="plot-histograms-of-total-counts-per-cell">6.1 Plot histograms of total counts per cell<a class="anchor" aria-label="anchor" href="#plot-histograms-of-total-counts-per-cell"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterDistributions.html">filterDistributions</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                    plot_type <span class="op">=</span> <span class="st">'hist'</span>,</span>
<span>                    detection <span class="op">=</span> <span class="st">'cells'</span>,</span>
<span>                    method <span class="op">=</span> <span class="st">'sum'</span>,</span>
<span>                    feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>                    nr_bins <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                                      save_name <span class="op">=</span> <span class="st">'3.1_totalexpr'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/filterDistributions.html">filterDistributions</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                    plot_type <span class="op">=</span> <span class="st">'hist'</span>,</span>
<span>                    detection <span class="op">=</span> <span class="st">'cells'</span>,</span>
<span>                    method <span class="op">=</span> <span class="st">'sum'</span>,</span>
<span>                    feat_type <span class="op">=</span> <span class="st">'neg_probe'</span>,</span>
<span>                    nr_bins <span class="op">=</span> <span class="fl">25</span>,</span>
<span>                    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                                      save_name <span class="op">=</span> <span class="st">'3.2_totalnegprbe'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>RNA</strong> <img src="images/nanostring_cosmx_lung_cancer/3.1_totalexpr.png"></p>
<p><strong>Negative Probes</strong> <img src="images/nanostring_cosmx_lung_cancer/3.2_totalnegprbe.png"></p>
</div>
<div class="section level3">
<h3 id="d-density-plots">6.2 2D Density Plots<a class="anchor" aria-label="anchor" href="#d-density-plots"></a>
</h3>
<p>Density-based representations may sometimes be preferred instead of
viewing the raw points information, especially when points are dense
enough that there is overplotting. After overlaps information has been
calculated, <code><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotDensity.html">spatInSituPlotDensity()</a></code> can be used in order to
get a general idea of how much expression there is of a feature.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotDensity.html">spatInSituPlotDensity</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                      feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MMP2"</span>, <span class="st">"VEGFA"</span>, <span class="st">"IGF1R"</span>,</span>
<span>                                <span class="st">'MKI67'</span>, <span class="st">'EPCAM'</span>, <span class="st">'KRT8'</span><span class="op">)</span>,</span>
<span>                      cow_n_col <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                      save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                                        save_name <span class="op">=</span> <span class="st">'4_inSituDens'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/4_inSituDens.png"></p>
</div>
<div class="section level3">
<h3 id="extract-data-from-giotto-object">6.3 Extract Data from Giotto Object<a class="anchor" aria-label="anchor" href="#extract-data-from-giotto-object"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># combine cell data</span></span>
<span><span class="va">morphometa</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/combineCellData.html">combineCellData</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                             feat_type <span class="op">=</span> <span class="st">'rna'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># combine feature data</span></span>
<span><span class="va">featmeta</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/combineFeatureData.html">combineFeatureData</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                              feat_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'rna'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># combine overlapping feature data</span></span>
<span><span class="va">featoverlapmeta</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/combineFeatureOverlapData.html">combineFeatureOverlapData</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                                            feat_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'rna'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="comparison-of-giotto-aggregated-and-nanostring-provided-matrices">6.4 Comparison of Giotto aggregated and Nanostring provided
matrices<a class="anchor" aria-label="anchor" href="#comparison-of-giotto-aggregated-and-nanostring-provided-matrices"></a>
</h3>
<p>Comparison of Giotto’s aggregated matrix results and those provided
by Nanostring. Only FOV2 will be used in this comparison. Matrices are
expected to be similar when the same sets of cell polygons/masks are
used for both.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load and prepare data</span></span>
<span></span>
<span><span class="va">nanoDT</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">'Lung12_exprMat_file.csv'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">test1</span> <span class="op">=</span> <span class="va">nanoDT</span><span class="op">[</span><span class="va">fov</span> <span class="op">==</span> <span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Set up cell_IDs</span></span>
<span><span class="va">test1</span><span class="op">[</span>, <span class="va">cell_ID</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'cell_'</span>, <span class="va">cell_ID</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">test1</span><span class="op">[</span>, <span class="va">cell_ID</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'f'</span>, <span class="va">fov</span>, <span class="st">'-'</span>, <span class="va">cell_ID</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">test1</span><span class="op">[</span>, <span class="va">fov</span> <span class="op">:=</span> <span class="cn">NULL</span><span class="op">]</span></span>
<span></span>
<span><span class="va">test1mat</span> <span class="op">=</span> <span class="fu">GiottoClass</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/t_flex.html">t_flex</a></span><span class="op">(</span><span class="fu">GiottoUtils</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/Giotto/reference/dt_to_matrix.html">dt_to_matrix</a></span><span class="op">(</span><span class="va">test1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">testnano_f2</span> <span class="op">=</span> <span class="va">test1mat</span></span>
<span></span>
<span><span class="co"># Remove cell_0 (all tx counts that do not fall within a polygon)</span></span>
<span><span class="va">testnano_f2</span> <span class="op">=</span> <span class="va">testnano_f2</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Remove negative probe counts</span></span>
<span><span class="va">testnano_f2</span> <span class="op">=</span> <span class="va">testnano_f2</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'NegPrb'</span>, <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">rownames</a></span><span class="op">(</span><span class="va">testnano_f2</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># giotto matrix</span></span>
<span><span class="va">testg</span> <span class="op">=</span> <span class="va">fov_join</span><span class="op">@</span><span class="va">expression</span><span class="op">$</span><span class="va">cell</span><span class="op">$</span><span class="va">rna</span><span class="op">$</span><span class="va">raw</span><span class="op">[</span><span class="op">]</span></span>
<span><span class="va">testg_f2</span> <span class="op">=</span> <span class="va">testg</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'fov002'</span>, <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">colnames</a></span><span class="op">(</span><span class="va">testg</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">sorted_rownames</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html">rownames</a></span><span class="op">(</span><span class="va">testg_f2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">testg_f2</span> <span class="op">=</span> <span class="va">testg_f2</span><span class="op">[</span><span class="va">sorted_rownames</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Prepare matrix comparison</span></span>
<span><span class="co"># Summarise sparse matrices (i and j are matrix indices, x is value)</span></span>
<span><span class="va">testg_f2_DT</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/as.data.table.html" class="external-link">as.data.table</a></span><span class="op">(</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu">summary</span><span class="op">(</span><span class="va">testg_f2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">testg_f2_DT</span><span class="op">[</span>, <span class="va">method</span> <span class="op">:=</span> <span class="st">'giotto'</span><span class="op">]</span></span>
<span><span class="va">testnano_f2_DT</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/as.data.table.html" class="external-link">as.data.table</a></span><span class="op">(</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu">summary</span><span class="op">(</span><span class="va">testnano_f2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">testnano_f2_DT</span><span class="op">[</span>, <span class="va">method</span> <span class="op">:=</span> <span class="st">'nanostring'</span><span class="op">]</span></span>
<span><span class="va">testDT</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/rbindlist.html" class="external-link">rbindlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">testg_f2_DT</span>, <span class="va">testnano_f2_DT</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine sparse matrix indices</span></span>
<span><span class="va">testDT</span><span class="op">[</span>, <span class="va">combo</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">i</span>,<span class="st">'-'</span>,<span class="va">j</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot results</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># matrix index similarity</span></span>
<span><span class="va">pl_n</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pl_n</span> <span class="op">=</span> <span class="va">pl_n</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">testnano_f2_DT</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">i</span>, y <span class="op">=</span> <span class="va">j</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">x</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pl_n</span> <span class="op">=</span> <span class="va">pl_n</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">'Nanostring Sparse Matrix'</span><span class="op">)</span></span>
<span><span class="va">pl_n</span> <span class="op">=</span> <span class="va">pl_n</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">'blue'</span>, high <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span>
<span><span class="va">pl_n</span> <span class="op">=</span> <span class="va">pl_n</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                    panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                    panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pl_g</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pl_g</span> <span class="op">=</span> <span class="va">pl_g</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">testg_f2_DT</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">i</span>, y <span class="op">=</span> <span class="va">j</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">x</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pl_g</span> <span class="op">=</span> <span class="va">pl_g</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">'Giotto Sparse Matrix'</span><span class="op">)</span></span>
<span><span class="va">pl_g</span> <span class="op">=</span> <span class="va">pl_g</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">'blue'</span>, high <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span>
<span><span class="va">pl_g</span> <span class="op">=</span> <span class="va">pl_g</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                    panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                    panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">combplot</span> <span class="op">=</span> <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">pl_n</span>, <span class="va">pl_g</span>,</span>
<span>                              nrow <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                              labels <span class="op">=</span> <span class="st">'AUTO'</span><span class="op">)</span></span>
<span><span class="va">combplot</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/mat_comparison.png"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># directly compare differences in matrix values (counts assigned)</span></span>
<span><span class="va">vartestDT</span> <span class="op">=</span> <span class="va">testDT</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>var <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">var</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, diff <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/diff.html" class="external-link">diff</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>, by <span class="op">=</span> <span class="fu">.</span><span class="op">(</span><span class="va">i</span>,<span class="va">j</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">setorder</a></span><span class="op">(</span><span class="va">vartestDT</span>, <span class="va">var</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check arbitrary index values</span></span>
<span><span class="va">testDT</span><span class="op">[</span><span class="va">i</span> <span class="op">==</span> <span class="st">'812'</span> <span class="op">&amp;</span> <span class="va">j</span> <span class="op">==</span> <span class="st">'2'</span><span class="op">]</span></span>
<span><span class="va">testDT</span><span class="op">[</span><span class="va">i</span> <span class="op">==</span> <span class="st">'667'</span> <span class="op">&amp;</span> <span class="va">j</span> <span class="op">==</span> <span class="st">'1072'</span><span class="op">]</span></span>
<span><span class="va">testDT</span><span class="op">[</span><span class="va">i</span> <span class="op">==</span> <span class="st">'667'</span> <span class="op">&amp;</span> <span class="va">j</span> <span class="op">==</span> <span class="st">'2880'</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># plot difference in values</span></span>
<span><span class="va">pl</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pl</span> <span class="op">=</span> <span class="va">pl</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">vartestDT</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">diff</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pl</span> <span class="op">=</span> <span class="va">pl</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pl</span> <span class="op">=</span> <span class="va">pl</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'difference nanostring - Giotto'</span><span class="op">)</span></span>
<span><span class="va">pl</span></span>
<span></span>
<span><span class="va">testDT</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/values_diff.png"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testDT</span><span class="op">[</span>, <span class="va">.N</span>, by <span class="op">=</span> <span class="st">'method'</span><span class="op">]</span></span>
<span></span>
<span><span class="va">testDT</span><span class="op">[</span>, <span class="va">method</span>, by <span class="op">=</span> <span class="va">combo</span><span class="op">]</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">combo</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#        method      N</span></span>
<span><span class="co"># 1:     giotto 415952</span></span>
<span><span class="co"># 2: nanostring 416099</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># 411050</span></span></code></pre></div>
<p>Overall, the nanostring matrix has <strong>416099 - 415952 =
147</strong> more non-zero values than giotto’s matrix for FOV2. Within
the <strong>411050</strong> shared entries that were called by both
methods (common i and j indices), there appears to be no major bias in
terms of counts/values assigned. Moreover, the vast majority of these
shared entries have the same values (difference of 0).</p>
</div>
</div>
<div class="section level2">
<h2 id="filtering-and-normalization">7. Filtering and normalization<a class="anchor" aria-label="anchor" href="#filtering-and-normalization"></a>
</h2>
<p>After the expression matrix is generated from the subcellular
information, analysis proceeds through data filtering and
normalization.</p>
<p>For the normalization step, we will employ two types.</p>
<ul>
<li><p><strong>standard normalization method:</strong> <em>library size
normalization and log normalization.</em> This method will produce both
normalized and scaled values that are be returned as the ‘normalized’
and ’scaled’expression matrices respectively. In this tutorial, the
normalized values will be used for generating expression statistics and
plotting expression values. The scaled values will be ignored. We will
also generate normalized values for the negative probes for
visualization purposes during which the library normalization step will
be skipped.</p></li>
<li><p><strong>pearson residuals:</strong> <em>A normalization that uses
the method described in</em> <a href="https://doi.org/10.1186/s13059-021-02451-7" class="external-link">Lause/Kobak et al,
2021</a>. This produces a set of values that are most similar in utility
to a scaled matrix and offer improvements to both HVF detection and PCA
generation. These values should not be used for statistics, plotting of
expression values, or differential expression analysis.</p></li>
</ul>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter (feat_type = 'rna' by default)</span></span>
<span><span class="va">fov_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                         feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>                         expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                         feat_det_in_min_cells <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                         min_det_feats_per_cell <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># normalize</span></span>
<span><span class="co"># standard method of normalization (log normalization based)</span></span>
<span><span class="va">fov_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                            feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>                            norm_methods <span class="op">=</span> <span class="st">'standard'</span>,</span>
<span>                            verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fov_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                            feat_type <span class="op">=</span> <span class="st">'neg_probe'</span>,</span>
<span>                            norm_methods <span class="op">=</span> <span class="st">'standard'</span>,</span>
<span>                            library_size_norm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># new normalization method based on pearson correlations (Lause/Kobak et al. 2021)</span></span>
<span><span class="co"># this normalized matrix is given the name 'pearson' using the update_slot param</span></span>
<span><span class="va">fov_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                            feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>                            scalefactor <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>                            verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            norm_methods <span class="op">=</span> <span class="st">'pearson_resid'</span>,</span>
<span>                            update_slot <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoExpression.html">showGiottoExpression</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add statistics based on log normalized values for features rna and negative probes</span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                         expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                         feat_type <span class="op">=</span> <span class="st">'rna'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                         expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                         feat_type <span class="op">=</span> <span class="st">'neg_probe'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View cellular data (default is feat = 'rna')</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoCellMetadata.html">showGiottoCellMetadata</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View feature data</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoFeatMetadata.html">showGiottoFeatMetadata</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span></code></pre></div>
<p><strong>Note:</strong> The show functions for metadata do not return
the information. To retrieve the metadata information, instead use
<code><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html">pDataDT()</a></code> and <code><a href="https://drieslab.github.io/GiottoClass/reference/fDataDT.html">fDataDT()</a></code> along with the
<code>feat_type</code> param for either ‘rna’ or ‘neg_probe’.</p>
</div>
<div class="section level2">
<h2 id="view-transcript-total-expression-distribution">8. View Transcript Total Expression Distribution<a class="anchor" aria-label="anchor" href="#view-transcript-total-expression-distribution"></a>
</h2>
<div class="section level3">
<h3 id="histogram-of-log-normalized-data">8.1 Histogram of log normalized data<a class="anchor" aria-label="anchor" href="#histogram-of-log-normalized-data"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterDistributions.html">filterDistributions</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                    detection <span class="op">=</span> <span class="st">'cells'</span>,</span>
<span>                    feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>                    expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                    method <span class="op">=</span> <span class="st">'sum'</span>,</span>
<span>                    nr_bins <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                                      save_name <span class="op">=</span> <span class="st">'5.1_rna_norm_total_hist'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/filterDistributions.html">filterDistributions</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                    detection <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                    feat_type <span class="op">=</span> <span class="st">'neg_probe'</span>,</span>
<span>                    expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                    method <span class="op">=</span> <span class="st">'sum'</span>,</span>
<span>                    nr_bins <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                                      save_name <span class="op">=</span> <span class="st">'5.2_neg_norm_total_hist'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>RNA</strong></p>
<p><img src="images/nanostring_cosmx_lung_cancer/5.1_rna_norm_total_hist.png"></p>
<p><strong>Negative Probe</strong></p>
<p><img src="images/nanostring_cosmx_lung_cancer/5.2_neg_norm_total_hist.png"></p>
</div>
<div class="section level3">
<h3 id="plot-spatially-as-centroids">8.2 Plot spatially as centroids<a class="anchor" aria-label="anchor" href="#plot-spatially-as-centroids"></a>
</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'total_expr'</span>,</span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           image_name <span class="op">=</span> <span class="va">image_names</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">0.9</span>,</span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.75</span>,</span>
<span>           save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                             save_name <span class="op">=</span> <span class="st">'5.3_color_centroids'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/5.3_color_centroids.png"></p>
</div>
<div class="section level3">
<h3 id="plot-spatially-as-color-scaled-polygons">8.3 Plot spatially as color-scaled polygons<a class="anchor" aria-label="anchor" href="#plot-spatially-as-color-scaled-polygons"></a>
</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">'gray'</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     polygon_fill <span class="op">=</span> <span class="st">'total_expr'</span>,</span>
<span>                     polygon_fill_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                       save_name <span class="op">=</span> <span class="st">'5.4_rna_color_polys'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                     feat_type <span class="op">=</span> <span class="st">'neg_probe'</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">'gray'</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     polygon_fill <span class="op">=</span> <span class="st">'total_expr'</span>,</span>
<span>                     polygon_fill_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                       save_name <span class="op">=</span> <span class="st">'5.5_neg_color_polys'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>RNA</strong></p>
<p><img src="images/nanostring_cosmx_lung_cancer/5.4_rna_color_polys.png"></p>
<p><strong>Negative Probe</strong></p>
<p><img src="images/nanostring_cosmx_lung_cancer/5.5_neg_color_polys.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="dimension-reduction">9. Dimension Reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<div class="section level3">
<h3 id="detect-highly-variable-genes-and-generate-pca">9.1 Detect highly variable genes and generate PCA<a class="anchor" aria-label="anchor" href="#detect-highly-variable-genes-and-generate-pca"></a>
</h3>
<p>Detect highly variable genes using the pearson residuals method based
on the ‘pearson’ expression matrix. These results will be returned as a
new ‘hvf’ column in the ‘rna’ feature metadata.</p>
<p>PCA generation will also be based on the ‘pearson’ matrix. Scaling
and centering of the PCA which is usually done by default will be
skipped since the pearson matrix is already scaled.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                        method <span class="op">=</span> <span class="st">'var_p_resid'</span>,</span>
<span>                        expression_values <span class="op">=</span> <span class="st">'pearson'</span>,</span>
<span>                        save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                                          save_name <span class="op">=</span> <span class="st">'6.1_pearson_HVF'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># print HVFs</span></span>
<span><span class="va">gene_meta</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/fDataDT.html">fDataDT</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span>
<span><span class="va">gene_meta</span><span class="op">[</span><span class="va">hvf</span> <span class="op">==</span> <span class="st">'yes'</span>, <span class="va">feat_ID</span><span class="op">]</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/6.1_pearson_HVF.png"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                  scale_unit <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                  center <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                  expression_values <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># screeplot uses the generated PCA. No need to specify expr values</span></span>
<span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">fov_join</span>, </span>
<span>          ncp <span class="op">=</span> <span class="fl">20</span>, </span>
<span>          save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'6.2_screeplot'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotPCA.html">plotPCA</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>        cell_color <span class="op">=</span> <span class="st">'nr_feats'</span>, <span class="co"># (from log norm statistics)</span></span>
<span>        color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        point_size <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>        point_shape <span class="op">=</span> <span class="st">'no_border'</span>,</span>
<span>        save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'6.3_PCA'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/6.2_screeplot.png"><img src="images/nanostring_cosmx_lung_cancer/6.3_PCA.png"></p>
</div>
<div class="section level3">
<h3 id="run-umap">9.2 Run UMAP<a class="anchor" aria-label="anchor" href="#run-umap"></a>
</h3>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate UMAP from PCA</span></span>
<span><span class="va">fov_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                    dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                    n_threads <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>, </span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'6.4_UMAP'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/6.4_UMAP.png"></p>
</div>
<div class="section level3">
<h3 id="plot-features-on-expression-space">9.3 Plot features on expression space<a class="anchor" aria-label="anchor" href="#plot-features-on-expression-space"></a>
</h3>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/dimFeatPlot2D.html">dimFeatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>              feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>              feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'MKI67'</span>, <span class="st">'CD8A'</span>, <span class="st">'CD4'</span>,</span>
<span>                        <span class="st">'COL1A1'</span>, <span class="st">'MS4A1'</span>, <span class="st">'MZB1'</span><span class="op">)</span>,</span>
<span>              expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>              point_shape <span class="op">=</span> <span class="st">'no_border'</span>,</span>
<span>              point_size <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>              cow_n_col <span class="op">=</span> <span class="fl">3</span>,</span>
<span>              save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                                save_name <span class="op">=</span> <span class="st">'6.5_UMAP_feats'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/6.5_UMAP_feats.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="cluster">10. Cluster<a class="anchor" aria-label="anchor" href="#cluster"></a>
</h2>
<div class="section level3">
<h3 id="visualize-clustering">10.1 Visualize clustering<a class="anchor" aria-label="anchor" href="#visualize-clustering"></a>
</h3>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fov_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                                 dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                                 k <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fov_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                            resolution <span class="op">=</span> <span class="fl">0.07</span>,</span>
<span>                            n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize UMAP cluster results</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>         cell_color_code <span class="op">=</span> <span class="va">pal10</span>,</span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'7.1_UMAP_leiden'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/7.1_UMAP_leiden.png"></p>
</div>
<div class="section level3">
<h3 id="visualize-clustering-on-expression-and-spatial-space">10.2 Visualize clustering on expression and spatial space<a class="anchor" aria-label="anchor" href="#visualize-clustering-on-expression-and-spatial-space"></a>
</h3>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># visualize UMAP and spatial results</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html">spatDimPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>              show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              image_name <span class="op">=</span> <span class="va">image_names</span>,</span>
<span>              cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>              cell_color_code <span class="op">=</span> <span class="va">pal10</span>,</span>
<span>              spat_point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>              save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'7.2_spatdim_leiden'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/7.2_spatdim_leiden.png"></p>
</div>
<div class="section level3">
<h3 id="map-clustering-spatially">10.3 Map clustering spatially<a class="anchor" aria-label="anchor" href="#map-clustering-spatially"></a>
</h3>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                     feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'rna'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'MMP2'</span>, <span class="st">'VEGFA'</span>, <span class="st">'IGF1R'</span>,</span>
<span>                                            <span class="st">'MKI67'</span>, <span class="st">'EPCAM'</span>, <span class="st">'MZB1'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                     point_size <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>                     feats_color_code <span class="op">=</span> <span class="va">viv10</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">'white'</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                     polygon_fill <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>                     polygon_fill_as_factor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_fill_code <span class="op">=</span> <span class="va">pal10</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                                       save_name <span class="op">=</span> <span class="st">'7.3_spatinsitu_leiden'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/7.3_spatinsitu_leiden.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="small-subset-visualization">11. Small Subset Visualization<a class="anchor" aria-label="anchor" href="#small-subset-visualization"></a>
</h2>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#subset a Giotto object based on spatial locations</span></span>
<span><span class="va">smallfov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiottoLocs.html">subsetGiottoLocs</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                             x_max <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>                             x_min <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>                             y_max <span class="op">=</span> <span class="op">-</span><span class="fl">157800</span>,</span>
<span>                             y_min <span class="op">=</span> <span class="op">-</span><span class="fl">159800</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#extract all genes observed in new object</span></span>
<span><span class="va">smallfeats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/fDataDT.html">fDataDT</a></span><span class="op">(</span><span class="va">smallfov</span><span class="op">)</span><span class="op">[</span>, <span class="va">feat_ID</span><span class="op">]</span></span>
<span></span>
<span><span class="co">#plot all genes</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">smallfov</span>,</span>
<span>                     feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">smallfeats</span><span class="op">)</span>,</span>
<span>                     point_size <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">'white'</span>,</span>
<span>                     show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     largeImage_name <span class="op">=</span> <span class="st">'fov002-composite'</span>,</span>
<span>                     show_legend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'8.1_smallfov_points'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot only the polygon outlines</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">smallfov</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                     polygon_alpha <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">'white'</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     largeImage_name <span class="op">=</span> <span class="st">'fov002-composite'</span>,</span>
<span>                     show_legend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'8.2_smallfov_poly'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot polygons colorlabeled with leiden clusters</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">smallfov</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_fill <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>                     polygon_fill_as_factor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_fill_code <span class="op">=</span> <span class="va">pal10</span>,</span>
<span>                     show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     largeImage_name <span class="op">=</span> <span class="st">'fov002-composite'</span>,</span>
<span>                     show_legend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'8.3_smallfov_leiden'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/8.1_smallfov_points.png"><img src="images/nanostring_cosmx_lung_cancer/8.2_smallfov_poly.png"><img src="images/nanostring_cosmx_lung_cancer/8.3_smallfov_leiden.png"></p>
</div>
<div class="section level2">
<h2 id="spatial-expression-patterns">12. Spatial Expression Patterns<a class="anchor" aria-label="anchor" href="#spatial-expression-patterns"></a>
</h2>
<p>Find spatially organized gene expression by examining the binarized
expression of cells and their spatial neighbors.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create spatial network based on physical distance of cell centroids</span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                                minimum_k <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                maximum_distance_delaunay <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># perform Binary Spatial Extraction of genes - NOTE: Depending on your system this could take time</span></span>
<span><span class="va">km_spatialgenes</span> <span class="op">=</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span><span class="va">fov_join</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize spatial expression of selected genes obtained from binSpect</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatFeatPlot2D.html">spatFeatPlot2D</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>               expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>               feats <span class="op">=</span> <span class="va">km_spatialgenes</span><span class="op">$</span><span class="va">feats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,</span>
<span>               point_shape <span class="op">=</span> <span class="st">'no_border'</span>,</span>
<span>               point_border_stroke <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>               point_size <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>               cow_n_col <span class="op">=</span> <span class="fl">2</span>,</span>
<span>               save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'9_binspect_genes'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/9_binspect_genes.png"></p>
</div>
<div class="section level2">
<h2 id="identify-cluster-differential-expression-genes">13. Identify cluster differential expression genes<a class="anchor" aria-label="anchor" href="#identify-cluster-differential-expression-genes"></a>
</h2>
<div class="section level3">
<h3 id="violin-plot">13.1 Violin plot<a class="anchor" aria-label="anchor" href="#violin-plot"></a>
</h3>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gini</span></span>
<span><span class="va">markers</span> <span class="op">=</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                                 method <span class="op">=</span> <span class="st">'gini'</span>,</span>
<span>                                 expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                                 cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>                                 min_feats <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                 rank_score <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co"># First 5 results by cluster</span></span>
<span><span class="va">markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">5</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># violinplot</span></span>
<span><span class="va">topgini_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/violinPlot.html">violinPlot</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>           feats <span class="op">=</span> <span class="va">topgini_genes</span>,</span>
<span>           cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>           strip_position <span class="op">=</span> <span class="st">'right'</span>,</span>
<span>           save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'10.1_gini_violin'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/10.1_gini_violin.png"></p>
</div>
<div class="section level3">
<h3 id="heatmap">13.2 Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cluster_order</span> <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotMetaDataHeatmap.html">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                    expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'leiden_clus'</span><span class="op">)</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="va">topgini_genes</span>,</span>
<span>                    custom_cluster_order <span class="op">=</span> <span class="va">cluster_order</span>,</span>
<span>                    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                                      save_name <span class="op">=</span> <span class="st">'10.2_heatmap'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/10.2_heatmap.png"></p>
</div>
<div class="section level3">
<h3 id="plot-gini-genes-on-umap">13.3 Plot gini genes on UMAP<a class="anchor" aria-label="anchor" href="#plot-gini-genes-on-umap"></a>
</h3>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># low, mid, high</span></span>
<span><span class="va">custom_scale</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'#440154'</span>, <span class="st">'#1F968B'</span>, <span class="st">'#FDE725'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/dimFeatPlot2D.html">dimFeatPlot2D</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>              expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>              cell_color_gradient <span class="op">=</span> <span class="va">custom_scale</span>,</span>
<span>              gradient_midpoint <span class="op">=</span> <span class="fl">5</span>,</span>
<span>              feats <span class="op">=</span> <span class="va">topgini_genes</span>,</span>
<span>              point_shape <span class="op">=</span> <span class="st">'no_border'</span>,</span>
<span>              point_size <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>              cow_n_col <span class="op">=</span> <span class="fl">4</span>,</span>
<span>              save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                                save_name <span class="op">=</span> <span class="st">'10.3_gini_genes'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/10.3_gini_genes.png"></p>
</div>
<div class="section level3">
<h3 id="annotate-giotto-object">13.4 Annotate Giotto Object<a class="anchor" aria-label="anchor" href="#annotate-giotto-object"></a>
</h3>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## add cell types ###</span></span>
<span><span class="va">clusters_cell_types_lung</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Normal Epithelial 1'</span>, <span class="st">'Cancer'</span>, <span class="st">'Stromal'</span>, <span class="st">'B-lineage'</span>,</span>
<span>                             <span class="st">'Macrophage'</span>, <span class="st">'B-lineage'</span>, <span class="st">'Cancer'</span>,</span>
<span>                             <span class="st">'Normal Epithelial 2'</span>, <span class="st">'Stromal'</span>, <span class="st">'B-lineage'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clusters_cell_types_lung</span><span class="op">)</span> <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span></span>
<span></span>
<span><span class="va">fov_join</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/annotateGiotto.html">annotateGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                          annotation_vector <span class="op">=</span> <span class="va">clusters_cell_types_lung</span>,</span>
<span>                          cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>                          name <span class="op">=</span> <span class="st">'cell_types'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>         cell_color_code <span class="op">=</span> <span class="va">viv10</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'11_anno_umap'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/11_anno_umap.png"></p>
</div>
<div class="section level3">
<h3 id="visualize">13.5 Visualize<a class="anchor" aria-label="anchor" href="#visualize"></a>
</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html">spatDimPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>              show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              image_name <span class="op">=</span> <span class="va">image_names</span>,</span>
<span>              cell_color <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>              cell_color_code <span class="op">=</span> <span class="va">viv10</span>,</span>
<span>              spat_point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>              save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'12_spatdim_type'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/12_spatdim_type.png"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_feat_type <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">'grey'</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     polygon_fill <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                     polygon_fill_as_factor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_fill_code <span class="op">=</span> <span class="va">viv10</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                       save_name <span class="op">=</span> <span class="st">'13_insitu_type'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/13_insitu_type.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="interaction-changed-genes">14. Interaction Changed Genes<a class="anchor" aria-label="anchor" href="#interaction-changed-genes"></a>
</h2>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">future</span><span class="fu">::</span><span class="fu"><a href="https://future.futureverse.org/reference/plan.html" class="external-link">plan</a></span><span class="op">(</span><span class="st">'multisession'</span>, workers <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="co"># NOTE: Depending on your system this could take time</span></span>
<span></span>
<span><span class="va">icf</span> <span class="op">=</span> <span class="fu"><a href="../reference/findInteractionChangedFeats.html">findInteractionChangedFeats</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>                                  cluster_column <span class="op">=</span> <span class="st">'cell_types'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify top ten interaction changed features</span></span>
<span><span class="va">icf</span><span class="op">$</span><span class="va">ICFscores</span><span class="op">[</span><span class="va">type_int</span> <span class="op">==</span> <span class="st">'hetero'</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Skip first two genes since they are too highly expressed</span></span>
<span><span class="va">icf_plotfeats</span> <span class="op">=</span> <span class="va">icf</span><span class="op">$</span><span class="va">ICFscores</span><span class="op">[</span><span class="va">type_int</span> <span class="op">==</span> <span class="st">'hetero'</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Visualize ICF expression</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">fov_join</span>,</span>
<span>                     feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">icf_plotfeats</span><span class="op">)</span>,</span>
<span>                     point_size <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_feat_type <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">'gray'</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     polygon_fill <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>                     polygon_fill_as_factor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_fill_code <span class="op">=</span> <span class="va">pal10</span>,</span>
<span>                     save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                                       save_name <span class="op">=</span> <span class="st">'14_ICF'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/nanostring_cosmx_lung_cancer/14_ICF.png"></p>
</div>
<div class="section level2">
<h2 id="saving-the-giotto-object">15. Saving the giotto object<a class="anchor" aria-label="anchor" href="#saving-the-giotto-object"></a>
</h2>
<p>Giotto uses many objects that include pointers to information that
live on disk instead of loading everything into memory. This includes
both giotto image objects (<code>giottoImage</code>,
<code>giottoLargeImage</code>) and also subcellular information
(<code>giottoPoints</code>, <code>giottoPolygon</code>). When saving the
project as a <code>.RDS</code> or <code>.Rdata</code>, these pointers
are broken and can produce errors when loaded again.</p>
<p><code><a href="https://drieslab.github.io/GiottoClass/reference/saveGiotto.html">saveGiotto()</a></code> is a function that can save Giotto Suite
projects into a contained structured directory that can then be properly
loaded again later using <code><a href="https://drieslab.github.io/GiottoClass/reference/loadGiotto.html">loadGiotto()</a></code>.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/saveGiotto.html">saveGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">fov_join</span>,</span>
<span>           foldername <span class="op">=</span> <span class="st">'new_folder_name'</span>,</span>
<span>           dir <span class="op">=</span> <span class="st">'/directory/to/save/to/'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">3.2</span> (<span class="dv">2023-10-31</span>)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>apple<span class="sc">-</span><span class="fu">darwin20</span> (<span class="dv">64</span><span class="sc">-</span>bit)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="fl">14.3</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.3</span><span class="sc">-</span>x86_64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] ggplot2_3.<span class="fl">4.4</span>     Giotto_4.<span class="fl">0.2</span>      GiottoClass_0.<span class="fl">1.3</span></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">1</span>] tidyselect_1.<span class="fl">2.0</span>            viridisLite_0.<span class="fl">4.2</span>           dplyr_1.<span class="fl">1.4</span>                </span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">4</span>] farver_2.<span class="fl">1.1</span>                GiottoVisuals_0.<span class="fl">1.4</span>         bitops_1<span class="fl">.0</span><span class="dv">-7</span>               </span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>  [<span class="dv">7</span>] fastmap_1.<span class="fl">1.1</span>               SingleCellExperiment_1.<span class="fl">24.0</span> RCurl_1<span class="fl">.98-1.14</span>            </span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a> [<span class="dv">10</span>] rsvd_1.<span class="fl">0.5</span>                  digest_0.<span class="fl">6.34</span>               lifecycle_1.<span class="fl">0.4</span>            </span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a> [<span class="dv">13</span>] terra_1<span class="fl">.7</span><span class="dv">-71</span>                dbscan_1<span class="fl">.1</span><span class="dv">-12</span>               magrittr_2.<span class="fl">0.3</span>             </span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a> [<span class="dv">16</span>] compiler_4.<span class="fl">3.2</span>              rlang_1.<span class="fl">1.3</span>                 tools_4.<span class="fl">3.2</span>                </span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a> [<span class="dv">19</span>] igraph_2.<span class="dv">0</span>.<span class="fl">1.1</span>              utf8_1.<span class="fl">2.4</span>                  yaml_2.<span class="fl">3.8</span>                 </span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a> [<span class="dv">22</span>] data.table_1.<span class="fl">15.0</span>           knitr_1<span class="fl">.45</span>                  S4Arrays_1.<span class="fl">2.0</span>             </span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a> [<span class="dv">25</span>] labeling_0.<span class="fl">4.3</span>              reticulate_1.<span class="fl">35.0</span>           DelayedArray_0.<span class="fl">28.0</span>        </span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a> [<span class="dv">28</span>] RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>          BiocParallel_1.<span class="fl">36.0</span>         abind_1<span class="fl">.4</span><span class="dv">-5</span>                </span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a> [<span class="dv">31</span>] withr_3.<span class="fl">0.0</span>                 BiocGenerics_0.<span class="fl">48.1</span>         grid_4.<span class="fl">3.2</span>                 </span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a> [<span class="dv">34</span>] stats4_4.<span class="fl">3.2</span>                fansi_1.<span class="fl">0.6</span>                 beachmat_2.<span class="fl">18.0</span>            </span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a> [<span class="dv">37</span>] colorspace_2<span class="fl">.1</span><span class="dv">-0</span>            future_1.<span class="fl">33.1</span>               progressr_0.<span class="fl">14.0</span>           </span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a> [<span class="dv">40</span>] globals_0.<span class="fl">16.2</span>              scales_1.<span class="fl">3.0</span>                gtools_3.<span class="fl">9.5</span>               </span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a> [<span class="dv">43</span>] MASS_7<span class="fl">.3</span><span class="dv">-60</span>.<span class="fl">0.1</span>             SummarizedExperiment_1.<span class="fl">32.0</span> cli_3.<span class="fl">6.2</span>                  </span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a> [<span class="dv">46</span>] rmarkdown_2<span class="fl">.25</span>              crayon_1.<span class="fl">5.2</span>                ragg_1.<span class="fl">2.7</span>                 </span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a> [<span class="dv">49</span>] generics_0.<span class="fl">1.3</span>              rstudioapi_0.<span class="fl">15.0</span>           future.apply_1.<span class="fl">11.1</span>        </span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a> [<span class="dv">52</span>] rjson_0.<span class="fl">2.21</span>                zlibbioc_1.<span class="fl">48.0</span>             parallel_4.<span class="fl">3.2</span>             </span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a> [<span class="dv">55</span>] XVector_0.<span class="fl">42.0</span>              matrixStats_1.<span class="fl">2.0</span>           vctrs_0.<span class="fl">6.5</span>                </span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a> [<span class="dv">58</span>] Matrix_1<span class="fl">.6</span><span class="dv">-5</span>                jsonlite_1.<span class="fl">8.8</span>              BiocSingular_1.<span class="fl">18.0</span>        </span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a> [<span class="dv">61</span>] IRanges_2.<span class="fl">36.0</span>              S4Vectors_0.<span class="fl">40.2</span>            ggrepel_0.<span class="fl">9.5</span>              </span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a> [<span class="dv">64</span>] irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>               scattermore_1<span class="fl">.2</span>             listenv_0.<span class="fl">9.1</span>              </span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a> [<span class="dv">67</span>] systemfonts_1.<span class="fl">0.5</span>           magick_2.<span class="fl">8.2</span>                GiottoUtils_0.<span class="fl">1.3</span>          </span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a> [<span class="dv">70</span>] parallelly_1.<span class="fl">36.0</span>           glue_1.<span class="fl">7.0</span>                  codetools_0<span class="fl">.2</span><span class="dv">-19</span>           </span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a> [<span class="dv">73</span>] uwot_0.<span class="fl">1.16</span>                 cowplot_1.<span class="fl">1.3</span>               RcppAnnoy_0.<span class="fl">0.22</span>           </span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a> [<span class="dv">76</span>] gtable_0.<span class="fl">3.4</span>                deldir_2<span class="fl">.0</span><span class="dv">-2</span>                GenomeInfoDb_1.<span class="fl">38.5</span>        </span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a> [<span class="dv">79</span>] GenomicRanges_1.<span class="fl">54.1</span>        ScaledMatrix_1.<span class="fl">10.0</span>         munsell_0.<span class="fl">5.0</span>              </span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a> [<span class="dv">82</span>] tibble_3.<span class="fl">2.1</span>                pillar_1.<span class="fl">9.0</span>                htmltools_0.<span class="fl">5.7</span>            </span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a> [<span class="dv">85</span>] GenomeInfoDbData_1.<span class="fl">2.11</span>     R6_2.<span class="fl">5.1</span>                    textshaping_0.<span class="fl">3.7</span>          </span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a> [<span class="dv">88</span>] evaluate_0<span class="fl">.23</span>               lattice_0<span class="fl">.22</span><span class="dv">-5</span>              Biobase_2.<span class="fl">62.0</span>             </span>
<span id="cb47-52"><a href="#cb47-52" aria-hidden="true" tabindex="-1"></a> [<span class="dv">91</span>] png_0<span class="fl">.1</span><span class="dv">-8</span>                   backports_1.<span class="fl">4.1</span>             SpatialExperiment_1.<span class="fl">12.0</span>   </span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a> [<span class="dv">94</span>] Rcpp_1.<span class="fl">0.12</span>                 SparseArray_1.<span class="fl">2.3</span>           checkmate_2.<span class="fl">3.1</span>            </span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a> [<span class="dv">97</span>] colorRamp2_0.<span class="fl">1.0</span>            xfun_0<span class="fl">.41</span>                   MatrixGenerics_1.<span class="fl">14.0</span>      </span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a>[<span class="dv">100</span>] pkgconfig_2.<span class="fl">0.3</span>  </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
